// astro-font-picker/dist/astroFontPicker.js
var me = { exports: {} };
(function(S) {
  (function() {
    function c(e, a, t) {
      return e.call.apply(e.bind, arguments);
    }
    function l(e, a, t) {
      if (!e)
        throw Error();
      if (2 < arguments.length) {
        var n = Array.prototype.slice.call(arguments, 2);
        return function() {
          var i = Array.prototype.slice.call(arguments);
          return Array.prototype.unshift.apply(i, n), e.apply(a, i);
        };
      }
      return function() {
        return e.apply(a, arguments);
      };
    }
    function f(e, a, t) {
      return f = Function.prototype.bind && Function.prototype.bind.toString().indexOf("native code") != -1 ? c : l, f.apply(null, arguments);
    }
    var v = Date.now || function() {
      return +/* @__PURE__ */ new Date();
    };
    function A(e, a) {
      this.a = e, this.o = a || e, this.c = this.o.document;
    }
    var h = !!window.FontFace;
    function k(e, a, t, n) {
      if (a = e.c.createElement(a), t)
        for (var i in t)
          t.hasOwnProperty(i) && (i == "style" ? a.style.cssText = t[i] : a.setAttribute(i, t[i]));
      return n && a.appendChild(e.c.createTextNode(n)), a;
    }
    function O(e, a, t) {
      e = e.c.getElementsByTagName(a)[0], e || (e = document.documentElement), e.insertBefore(t, e.lastChild);
    }
    function N(e) {
      e.parentNode && e.parentNode.removeChild(e);
    }
    function D(e, a, t) {
      a = a || [], t = t || [];
      for (var n = e.className.split(/\s+/), i = 0; i < a.length; i += 1) {
        for (var r = !1, p = 0; p < n.length; p += 1)
          if (a[i] === n[p]) {
            r = !0;
            break;
          }
        r || n.push(a[i]);
      }
      for (a = [], i = 0; i < n.length; i += 1) {
        for (r = !1, p = 0; p < t.length; p += 1)
          if (n[i] === t[p]) {
            r = !0;
            break;
          }
        r || a.push(n[i]);
      }
      e.className = a.join(" ").replace(/\s+/g, " ").replace(/^\s+|\s+$/, "");
    }
    function B(e, a) {
      for (var t = e.className.split(/\s+/), n = 0, i = t.length; n < i; n++)
        if (t[n] == a)
          return !0;
      return !1;
    }
    function b(e) {
      return e.o.location.hostname || e.a.location.hostname;
    }
    function R(e, a, t) {
      function n() {
        y && i && r && (y(p), y = null);
      }
      a = k(e, "link", { rel: "stylesheet", href: a, media: "all" });
      var i = !1, r = !0, p = null, y = t || null;
      h ? (a.onload = function() {
        i = !0, n();
      }, a.onerror = function() {
        i = !0, p = Error("Stylesheet failed to load"), n();
      }) : setTimeout(function() {
        i = !0, n();
      }, 0), O(e, "head", a);
    }
    function L(e, a, t, n) {
      var i = e.c.getElementsByTagName("head")[0];
      if (i) {
        var r = k(e, "script", { src: a }), p = !1;
        return r.onload = r.onreadystatechange = function() {
          p || this.readyState && this.readyState != "loaded" && this.readyState != "complete" || (p = !0, t && t(null), r.onload = r.onreadystatechange = null, r.parentNode.tagName == "HEAD" && i.removeChild(r));
        }, i.appendChild(r), setTimeout(function() {
          p || (p = !0, t && t(Error("Script load timeout")));
        }, n || 5e3), r;
      }
      return null;
    }
    function P() {
      this.a = 0, this.c = null;
    }
    function d(e) {
      return e.a++, function() {
        e.a--, G(e);
      };
    }
    function C(e, a) {
      e.c = a, G(e);
    }
    function G(e) {
      e.a == 0 && e.c && (e.c(), e.c = null);
    }
    function V(e) {
      this.a = e || "-";
    }
    V.prototype.c = function(e) {
      for (var a = [], t = 0; t < arguments.length; t++)
        a.push(arguments[t].replace(/[\W_]+/g, "").toLowerCase());
      return a.join(this.a);
    };
    function w(e, a) {
      this.c = e, this.f = 4, this.a = "n";
      var t = (a || "n4").match(/^([nio])([1-9])$/i);
      t && (this.a = t[1], this.f = parseInt(t[2], 10));
    }
    function le(e) {
      return J(e) + " " + (e.f + "00") + " 300px " + Y(e.c);
    }
    function Y(e) {
      var a = [];
      e = e.split(/,\s*/);
      for (var t = 0; t < e.length; t++) {
        var n = e[t].replace(/['"]/g, "");
        n.indexOf(" ") != -1 || /^\d/.test(n) ? a.push("'" + n + "'") : a.push(n);
      }
      return a.join(",");
    }
    function g(e) {
      return e.a + e.f;
    }
    function J(e) {
      var a = "normal";
      return e.a === "o" ? a = "oblique" : e.a === "i" && (a = "italic"), a;
    }
    function fe(e) {
      var a = 4, t = "n", n = null;
      return e && ((n = e.match(/(normal|oblique|italic)/i)) && n[1] && (t = n[1].substr(0, 1).toLowerCase()), (n = e.match(/([1-9]00|normal|bold)/i)) && n[1] && (/bold/i.test(n[1]) ? a = 7 : /[1-9]00/.test(n[1]) && (a = parseInt(n[1].substr(0, 1), 10)))), t + a;
    }
    function de(e, a) {
      this.c = e, this.f = e.o.document.documentElement, this.h = a, this.a = new V("-"), this.j = a.events !== !1, this.g = a.classes !== !1;
    }
    function ue(e) {
      e.g && D(e.f, [e.a.c("wf", "loading")]), T(e, "loading");
    }
    function Z(e) {
      if (e.g) {
        var a = B(e.f, e.a.c("wf", "active")), t = [], n = [e.a.c("wf", "loading")];
        a || t.push(e.a.c("wf", "inactive")), D(e.f, t, n);
      }
      T(e, "inactive");
    }
    function T(e, a, t) {
      e.j && e.h[a] && (t ? e.h[a](t.c, g(t)) : e.h[a]());
    }
    function ce() {
      this.c = {};
    }
    function ge(e, a, t) {
      var n = [], i;
      for (i in a)
        if (a.hasOwnProperty(i)) {
          var r = e.c[i];
          r && n.push(r(a[i], t));
        }
      return n;
    }
    function F(e, a) {
      this.c = e, this.f = a, this.a = k(this.c, "span", { "aria-hidden": "true" }, this.f);
    }
    function E(e) {
      O(e.c, "body", e.a);
    }
    function K(e) {
      return "display:block;position:absolute;top:-9999px;left:-9999px;font-size:300px;width:auto;height:auto;line-height:normal;margin:0;padding:0;font-variant:normal;white-space:nowrap;font-family:" + Y(e.c) + ";" + ("font-style:" + J(e) + ";font-weight:" + (e.f + "00") + ";");
    }
    function _(e, a, t, n, i, r) {
      this.g = e, this.j = a, this.a = n, this.c = t, this.f = i || 3e3, this.h = r || void 0;
    }
    _.prototype.start = function() {
      var e = this.c.o.document, a = this, t = v(), n = new Promise(function(p, y) {
        function s() {
          v() - t >= a.f ? y() : e.fonts.load(le(a.a), a.h).then(function(o) {
            1 <= o.length ? p() : setTimeout(s, 25);
          }, function() {
            y();
          });
        }
        s();
      }), i = null, r = new Promise(function(p, y) {
        i = setTimeout(y, a.f);
      });
      Promise.race([r, n]).then(function() {
        i && (clearTimeout(i), i = null), a.g(a.a);
      }, function() {
        a.j(a.a);
      });
    };
    function Q(e, a, t, n, i, r, p) {
      this.v = e, this.B = a, this.c = t, this.a = n, this.s = p || "BESbswy", this.f = {}, this.w = i || 3e3, this.u = r || null, this.m = this.j = this.h = this.g = null, this.g = new F(this.c, this.s), this.h = new F(this.c, this.s), this.j = new F(this.c, this.s), this.m = new F(this.c, this.s), e = new w(this.a.c + ",serif", g(this.a)), e = K(e), this.g.a.style.cssText = e, e = new w(this.a.c + ",sans-serif", g(this.a)), e = K(e), this.h.a.style.cssText = e, e = new w("serif", g(this.a)), e = K(e), this.j.a.style.cssText = e, e = new w("sans-serif", g(this.a)), e = K(e), this.m.a.style.cssText = e, E(this.g), E(this.h), E(this.j), E(this.m);
    }
    var I = { D: "serif", C: "sans-serif" }, W = null;
    function $() {
      if (W === null) {
        var e = /AppleWebKit\/([0-9]+)(?:\.([0-9]+))/.exec(window.navigator.userAgent);
        W = !!e && (536 > parseInt(e[1], 10) || parseInt(e[1], 10) === 536 && 11 >= parseInt(e[2], 10));
      }
      return W;
    }
    Q.prototype.start = function() {
      this.f.serif = this.j.a.offsetWidth, this.f["sans-serif"] = this.m.a.offsetWidth, this.A = v(), ee(this);
    };
    function X(e, a, t) {
      for (var n in I)
        if (I.hasOwnProperty(n) && a === e.f[I[n]] && t === e.f[I[n]])
          return !0;
      return !1;
    }
    function ee(e) {
      var a = e.g.a.offsetWidth, t = e.h.a.offsetWidth, n;
      (n = a === e.f.serif && t === e.f["sans-serif"]) || (n = $() && X(e, a, t)), n ? v() - e.A >= e.w ? $() && X(e, a, t) && (e.u === null || e.u.hasOwnProperty(e.a.c)) ? z(e, e.v) : z(e, e.B) : he(e) : z(e, e.v);
    }
    function he(e) {
      setTimeout(f(function() {
        ee(this);
      }, e), 50);
    }
    function z(e, a) {
      setTimeout(f(function() {
        N(this.g.a), N(this.h.a), N(this.j.a), N(this.m.a), a(this.a);
      }, e), 0);
    }
    function U(e, a, t) {
      this.c = e, this.a = a, this.f = 0, this.m = this.j = !1, this.s = t;
    }
    var j = null;
    U.prototype.g = function(e) {
      var a = this.a;
      a.g && D(a.f, [a.a.c("wf", e.c, g(e).toString(), "active")], [a.a.c("wf", e.c, g(e).toString(), "loading"), a.a.c("wf", e.c, g(e).toString(), "inactive")]), T(a, "fontactive", e), this.m = !0, ae(this);
    }, U.prototype.h = function(e) {
      var a = this.a;
      if (a.g) {
        var t = B(a.f, a.a.c("wf", e.c, g(e).toString(), "active")), n = [], i = [a.a.c("wf", e.c, g(e).toString(), "loading")];
        t || n.push(a.a.c("wf", e.c, g(e).toString(), "inactive")), D(a.f, n, i);
      }
      T(a, "fontinactive", e), ae(this);
    };
    function ae(e) {
      --e.f == 0 && e.j && (e.m ? (e = e.a, e.g && D(e.f, [e.a.c("wf", "active")], [e.a.c("wf", "loading"), e.a.c("wf", "inactive")]), T(e, "active")) : Z(e.a));
    }
    function ne(e) {
      this.j = e, this.a = new ce(), this.h = 0, this.f = this.g = !0;
    }
    ne.prototype.load = function(e) {
      this.c = new A(this.j, e.context || this.j), this.g = e.events !== !1, this.f = e.classes !== !1, we(this, new de(this.c, e), e);
    };
    function De(e, a, t, n, i) {
      var r = --e.h == 0;
      (e.f || e.g) && setTimeout(function() {
        var p = i || null, y = n || null || {};
        if (t.length === 0 && r)
          Z(a.a);
        else {
          a.f += t.length, r && (a.j = r);
          var s, o = [];
          for (s = 0; s < t.length; s++) {
            var m = t[s], u = y[m.c], H = a.a, x = m;
            if (H.g && D(H.f, [H.a.c("wf", x.c, g(x).toString(), "loading")]), T(H, "fontloading", x), H = null, j === null)
              if (window.FontFace) {
                var x = /Gecko.*Firefox\/(\d+)/.exec(window.navigator.userAgent), Le = /OS X.*Version\/10\..*Safari/.exec(window.navigator.userAgent) && /Apple/.exec(window.navigator.vendor);
                j = x ? 42 < parseInt(x[1], 10) : !Le;
              } else
                j = !1;
            j ? H = new _(f(a.g, a), f(a.h, a), a.c, m, a.s, u) : H = new Q(f(a.g, a), f(a.h, a), a.c, m, a.s, p, u), o.push(H);
          }
          for (s = 0; s < o.length; s++)
            o[s].start();
        }
      }, 0);
    }
    function we(e, a, t) {
      var i = [], n = t.timeout;
      ue(a);
      var i = ge(e.a, t, e.c), r = new U(e.c, a, n);
      for (e.h = i.length, a = 0, t = i.length; a < t; a++)
        i[a].load(function(p, y, s) {
          De(e, r, p, y, s);
        });
    }
    function te(e, a) {
      this.c = e, this.a = a;
    }
    te.prototype.load = function(e) {
      function a() {
        if (r["__mti_fntLst" + n]) {
          var p = r["__mti_fntLst" + n](), y = [], s;
          if (p)
            for (var o = 0; o < p.length; o++) {
              var m = p[o].fontfamily;
              p[o].fontStyle != null && p[o].fontWeight != null ? (s = p[o].fontStyle + p[o].fontWeight, y.push(new w(m, s))) : y.push(new w(m));
            }
          e(y);
        } else
          setTimeout(function() {
            a();
          }, 50);
      }
      var t = this, n = t.a.projectId, i = t.a.version;
      if (n) {
        var r = t.c.o;
        L(this.c, (t.a.api || "https://fast.fonts.net/jsapi") + "/" + n + ".js" + (i ? "?v=" + i : ""), function(p) {
          p ? e([]) : (r["__MonotypeConfiguration__" + n] = function() {
            return t.a;
          }, a());
        }).id = "__MonotypeAPIScript__" + n;
      } else
        e([]);
    };
    function ie(e, a) {
      this.c = e, this.a = a;
    }
    ie.prototype.load = function(e) {
      var a, t, n = this.a.urls || [], i = this.a.families || [], r = this.a.testStrings || {}, p = new P();
      for (a = 0, t = n.length; a < t; a++)
        R(this.c, n[a], d(p));
      var y = [];
      for (a = 0, t = i.length; a < t; a++)
        if (n = i[a].split(":"), n[1])
          for (var s = n[1].split(","), o = 0; o < s.length; o += 1)
            y.push(new w(n[0], s[o]));
        else
          y.push(new w(n[0]));
      C(p, function() {
        e(y, r);
      });
    };
    function He(e, a) {
      e ? this.c = e : this.c = Ne, this.a = [], this.f = [], this.g = a || "";
    }
    var Ne = "https://fonts.googleapis.com/css";
    function Me(e, a) {
      for (var t = a.length, n = 0; n < t; n++) {
        var i = a[n].split(":");
        i.length == 3 && e.f.push(i.pop());
        var r = "";
        i.length == 2 && i[1] != "" && (r = ":"), e.a.push(i.join(r));
      }
    }
    function ve(e) {
      if (e.a.length == 0)
        throw Error("No fonts to load!");
      if (e.c.indexOf("kit=") != -1)
        return e.c;
      for (var a = e.a.length, t = [], n = 0; n < a; n++)
        t.push(e.a[n].replace(/ /g, "+"));
      return a = e.c + "?family=" + t.join("%7C"), 0 < e.f.length && (a += "&subset=" + e.f.join(",")), 0 < e.g.length && (a += "&text=" + encodeURIComponent(e.g)), a;
    }
    function ke(e) {
      this.f = e, this.a = [], this.c = {};
    }
    var re = { latin: "BESbswy", "latin-ext": "çöüğş", cyrillic: "йяЖ", greek: "αβΣ", khmer: "កខគ", Hanuman: "កខគ" }, be = { thin: "1", extralight: "2", "extra-light": "2", ultralight: "2", "ultra-light": "2", light: "3", regular: "4", book: "4", medium: "5", "semi-bold": "6", semibold: "6", "demi-bold": "6", demibold: "6", bold: "7", "extra-bold": "8", extrabold: "8", "ultra-bold": "8", ultrabold: "8", black: "9", heavy: "9", l: "3", r: "4", b: "7" }, Ce = { i: "i", italic: "i", n: "n", normal: "n" }, Ae = /^(thin|(?:(?:extra|ultra)-?)?light|regular|book|medium|(?:(?:semi|demi|extra|ultra)-?)?bold|black|heavy|l|r|b|[1-9]00)?(n|i|normal|italic)?$/;
    function Oe(e) {
      for (var a = e.f.length, t = 0; t < a; t++) {
        var n = e.f[t].split(":"), i = n[0].replace(/\+/g, " "), r = ["n4"];
        if (2 <= n.length) {
          var p, y = n[1];
          if (p = [], y)
            for (var y = y.split(","), s = y.length, o = 0; o < s; o++) {
              var m;
              if (m = y[o], m.match(/^[\w-]+$/)) {
                var u = Ae.exec(m.toLowerCase());
                if (u == null)
                  m = "";
                else {
                  if (m = u[2], m = m == null || m == "" ? "n" : Ce[m], u = u[1], u == null || u == "")
                    u = "4";
                  else
                    var H = be[u], u = H || (isNaN(u) ? "4" : u.substr(0, 1));
                  m = [m, u].join("");
                }
              } else
                m = "";
              m && p.push(m);
            }
          0 < p.length && (r = p), n.length == 3 && (n = n[2], p = [], n = n ? n.split(",") : p, 0 < n.length && (n = re[n[0]]) && (e.c[i] = n));
        }
        for (e.c[i] || (n = re[i]) && (e.c[i] = n), n = 0; n < r.length; n += 1)
          e.a.push(new w(i, r[n]));
      }
    }
    function pe(e, a) {
      this.c = e, this.a = a;
    }
    var Be = { Arimo: !0, Cousine: !0, Tinos: !0 };
    pe.prototype.load = function(e) {
      var a = new P(), t = this.c, n = new He(this.a.api, this.a.text), i = this.a.families;
      Me(n, i);
      var r = new ke(i);
      Oe(r), R(t, ve(n), d(a)), C(a, function() {
        e(r.a, r.c, Be);
      });
    };
    function ye(e, a) {
      this.c = e, this.a = a;
    }
    ye.prototype.load = function(e) {
      var a = this.a.id, t = this.c.o;
      a ? L(this.c, (this.a.api || "https://use.typekit.net") + "/" + a + ".js", function(n) {
        if (n)
          e([]);
        else if (t.Typekit && t.Typekit.config && t.Typekit.config.fn) {
          n = t.Typekit.config.fn;
          for (var i = [], r = 0; r < n.length; r += 2)
            for (var p = n[r], y = n[r + 1], s = 0; s < y.length; s++)
              i.push(new w(p, y[s]));
          try {
            t.Typekit.load({ events: !1, classes: !1, async: !0 });
          } catch {
          }
          e(i);
        }
      }, 2e3) : e([]);
    };
    function se(e, a) {
      this.c = e, this.f = a, this.a = [];
    }
    se.prototype.load = function(e) {
      var a = this.f.id, t = this.c.o, n = this;
      a ? (t.__webfontfontdeckmodule__ || (t.__webfontfontdeckmodule__ = {}), t.__webfontfontdeckmodule__[a] = function(i, r) {
        for (var p = 0, y = r.fonts.length; p < y; ++p) {
          var s = r.fonts[p];
          n.a.push(new w(s.name, fe("font-weight:" + s.weight + ";font-style:" + s.style)));
        }
        e(n.a);
      }, L(this.c, (this.f.api || "https://f.fontdeck.com/s/css/js/") + b(this.c) + "/" + a + ".js", function(i) {
        i && e([]);
      })) : e([]);
    };
    var M = new ne(window);
    M.a.c.custom = function(e, a) {
      return new ie(a, e);
    }, M.a.c.fontdeck = function(e, a) {
      return new se(a, e);
    }, M.a.c.monotype = function(e, a) {
      return new te(a, e);
    }, M.a.c.typekit = function(e, a) {
      return new ye(a, e);
    }, M.a.c.google = function(e, a) {
      return new pe(a, e);
    };
    var Se = { load: f(M.load, M) };
    S.exports ? S.exports = Se : (window.WebFont = Se, window.WebFontConfig && M.load(window.WebFontConfig));
  })();
})(me);
var Pe = me.exports;
const Te = `
  <h1>
    <span>Font Picker</span>
    <span class="toggle-label">Enabled</span>
    <astro-dev-toolbar-toggle id="toggle"></astro-dev-toolbar-toggle>
    <span class="toggle-label">Override All Styles</span>
    <astro-dev-toolbar-toggle id="override"></astro-dev-toolbar-toggle>
  </h1>
  <div class="flex">
    <select id="group-select"></select>
    <select id="font-select"></select>    
    <a id="google-link" href="" target="_blank">View in Google Fonts</a>
  </div>
  <div class="instr">
    <div id="instr1"></div>
    <br/>
    <div id="instr2"></div>
  </div>
  
  <style>
    
    h1 {
        margin: 0;
        display: inline-block;
      }
      
      .toggle-label {
        margin-left: 20px;
        font-size: 14px;
      }
      
      .flex {
        display: flex;
        align-items: baseline;
      }
      
      select {
        margin-top: 10px;
        padding: 10px;
        border-radius: 5px;
        border: none;
        background-color: #222;
        color: #eee;
      }
      
      select:focus-visible {
        outline: 1px solid #888;
      }
      
      select + select {
        margin-left: 10px;
      }
      
      select option {
        background-color: #333;
        color: #eee;
        padding: 5px 5px;
      }
      
      .instr {
        margin-top: 10px;
        border-radius: 5px;
        overflow: hidden;
        padding: 10px;
        background-color: #333;
        font-size: 12px;
        font-family: monospace;
      }
      
      a {
        color: inherit;
        margin-left: 20px;
      }
</style>
`, xe = [
  {
    name: "ABeeZee",
    type: "Sans Serif"
  },
  {
    name: "ADLaM Display",
    type: "Display"
  },
  {
    name: "AR One Sans",
    type: "Sans Serif"
  },
  {
    name: "Abel",
    type: "Sans Serif"
  },
  {
    name: "Abhaya Libre",
    type: "Serif"
  },
  {
    name: "Aboreto",
    type: "Display"
  },
  {
    name: "Abril Fatface",
    type: "Display"
  },
  {
    name: "Abyssinica SIL",
    type: "Serif"
  },
  {
    name: "Aclonica",
    type: "Sans Serif"
  },
  {
    name: "Acme",
    type: "Sans Serif"
  },
  {
    name: "Actor",
    type: "Sans Serif"
  },
  {
    name: "Adamina",
    type: "Serif"
  },
  {
    name: "Advent Pro",
    type: "Sans Serif"
  },
  {
    name: "Afacad",
    type: "Sans Serif"
  },
  {
    name: "Agbalumo",
    type: "Display"
  },
  {
    name: "Agdasima",
    type: "Sans Serif"
  },
  {
    name: "Aguafina Script",
    type: "Handwriting"
  },
  {
    name: "Akatab",
    type: "Sans Serif"
  },
  {
    name: "Akaya Kanadaka",
    type: "Display"
  },
  {
    name: "Akaya Telivigala",
    type: "Display"
  },
  {
    name: "Akronim",
    type: "Display"
  },
  {
    name: "Akshar",
    type: "Sans Serif"
  },
  {
    name: "Aladin",
    type: "Display"
  },
  {
    name: "Alata",
    type: "Sans Serif"
  },
  {
    name: "Alatsi",
    type: "Sans Serif"
  },
  {
    name: "Albert Sans",
    type: "Sans Serif"
  },
  {
    name: "Aldrich",
    type: "Sans Serif"
  },
  {
    name: "Alef",
    type: "Sans Serif"
  },
  {
    name: "Alegreya",
    type: "Serif"
  },
  {
    name: "Alegreya SC",
    type: "Serif"
  },
  {
    name: "Alegreya Sans",
    type: "Sans Serif"
  },
  {
    name: "Alegreya Sans SC",
    type: "Sans Serif"
  },
  {
    name: "Aleo",
    type: "Serif"
  },
  {
    name: "Alex Brush",
    type: "Handwriting"
  },
  {
    name: "Alexandria",
    type: "Sans Serif"
  },
  {
    name: "Alfa Slab One",
    type: "Display"
  },
  {
    name: "Alice",
    type: "Serif"
  },
  {
    name: "Alike",
    type: "Serif"
  },
  {
    name: "Alike Angular",
    type: "Serif"
  },
  {
    name: "Alkalami",
    type: "Serif"
  },
  {
    name: "Alkatra",
    type: "Display"
  },
  {
    name: "Allan",
    type: "Display"
  },
  {
    name: "Allerta",
    type: "Sans Serif"
  },
  {
    name: "Allerta Stencil",
    type: "Sans Serif"
  },
  {
    name: "Allison",
    type: "Handwriting"
  },
  {
    name: "Allura",
    type: "Handwriting"
  },
  {
    name: "Almarai",
    type: "Sans Serif"
  },
  {
    name: "Almendra",
    type: "Serif"
  },
  {
    name: "Almendra Display",
    type: "Display"
  },
  {
    name: "Almendra SC",
    type: "Serif"
  },
  {
    name: "Alumni Sans",
    type: "Sans Serif"
  },
  {
    name: "Alumni Sans Collegiate One",
    type: "Sans Serif"
  },
  {
    name: "Alumni Sans Inline One",
    type: "Display"
  },
  {
    name: "Alumni Sans Pinstripe",
    type: "Sans Serif"
  },
  {
    name: "Amarante",
    type: "Display"
  },
  {
    name: "Amaranth",
    type: "Sans Serif"
  },
  {
    name: "Amatic SC",
    type: "Handwriting"
  },
  {
    name: "Amethysta",
    type: "Serif"
  },
  {
    name: "Amiko",
    type: "Sans Serif"
  },
  {
    name: "Amiri",
    type: "Serif"
  },
  {
    name: "Amiri Quran",
    type: "Serif"
  },
  {
    name: "Amita",
    type: "Handwriting"
  },
  {
    name: "Anaheim",
    type: "Sans Serif"
  },
  {
    name: "Andada Pro",
    type: "Serif"
  },
  {
    name: "Andika",
    type: "Sans Serif"
  },
  {
    name: "Anek Bangla",
    type: "Sans Serif"
  },
  {
    name: "Anek Devanagari",
    type: "Sans Serif"
  },
  {
    name: "Anek Gujarati",
    type: "Sans Serif"
  },
  {
    name: "Anek Gurmukhi",
    type: "Sans Serif"
  },
  {
    name: "Anek Kannada",
    type: "Sans Serif"
  },
  {
    name: "Anek Latin",
    type: "Sans Serif"
  },
  {
    name: "Anek Malayalam",
    type: "Sans Serif"
  },
  {
    name: "Anek Odia",
    type: "Sans Serif"
  },
  {
    name: "Anek Tamil",
    type: "Sans Serif"
  },
  {
    name: "Anek Telugu",
    type: "Sans Serif"
  },
  {
    name: "Angkor",
    type: "Display"
  },
  {
    name: "Annie Use Your Telescope",
    type: "Handwriting"
  },
  {
    name: "Anonymous Pro",
    type: "Monospace"
  },
  {
    name: "Antic",
    type: "Sans Serif"
  },
  {
    name: "Antic Didone",
    type: "Serif"
  },
  {
    name: "Antic Slab",
    type: "Serif"
  },
  {
    name: "Anton",
    type: "Sans Serif"
  },
  {
    name: "Antonio",
    type: "Sans Serif"
  },
  {
    name: "Anuphan",
    type: "Sans Serif"
  },
  {
    name: "Anybody",
    type: "Display"
  },
  {
    name: "Aoboshi One",
    type: "Serif"
  },
  {
    name: "Arapey",
    type: "Serif"
  },
  {
    name: "Arbutus",
    type: "Serif"
  },
  {
    name: "Arbutus Slab",
    type: "Serif"
  },
  {
    name: "Architects Daughter",
    type: "Handwriting"
  },
  {
    name: "Archivo",
    type: "Sans Serif"
  },
  {
    name: "Archivo Black",
    type: "Sans Serif"
  },
  {
    name: "Archivo Narrow",
    type: "Sans Serif"
  },
  {
    name: "Are You Serious",
    type: "Handwriting"
  },
  {
    name: "Aref Ruqaa",
    type: "Serif"
  },
  {
    name: "Aref Ruqaa Ink",
    type: "Serif"
  },
  {
    name: "Arima",
    type: "Display"
  },
  {
    name: "Arimo",
    type: "Sans Serif"
  },
  {
    name: "Arizonia",
    type: "Handwriting"
  },
  {
    name: "Armata",
    type: "Sans Serif"
  },
  {
    name: "Arsenal",
    type: "Sans Serif"
  },
  {
    name: "Artifika",
    type: "Serif"
  },
  {
    name: "Arvo",
    type: "Serif"
  },
  {
    name: "Arya",
    type: "Sans Serif"
  },
  {
    name: "Asap",
    type: "Sans Serif"
  },
  {
    name: "Asap Condensed",
    type: "Sans Serif"
  },
  {
    name: "Asar",
    type: "Serif"
  },
  {
    name: "Asset",
    type: "Display"
  },
  {
    name: "Assistant",
    type: "Sans Serif"
  },
  {
    name: "Astloch",
    type: "Display"
  },
  {
    name: "Asul",
    type: "Sans Serif"
  },
  {
    name: "Athiti",
    type: "Sans Serif"
  },
  {
    name: "Atkinson Hyperlegible",
    type: "Sans Serif"
  },
  {
    name: "Atma",
    type: "Display"
  },
  {
    name: "Atomic Age",
    type: "Display"
  },
  {
    name: "Aubrey",
    type: "Display"
  },
  {
    name: "Audiowide",
    type: "Display"
  },
  {
    name: "Autour One",
    type: "Display"
  },
  {
    name: "Average",
    type: "Serif"
  },
  {
    name: "Average Sans",
    type: "Sans Serif"
  },
  {
    name: "Averia Gruesa Libre",
    type: "Display"
  },
  {
    name: "Averia Libre",
    type: "Display"
  },
  {
    name: "Averia Sans Libre",
    type: "Display"
  },
  {
    name: "Averia Serif Libre",
    type: "Display"
  },
  {
    name: "Azeret Mono",
    type: "Monospace"
  },
  {
    name: "B612",
    type: "Sans Serif"
  },
  {
    name: "B612 Mono",
    type: "Monospace"
  },
  {
    name: "BIZ UDGothic",
    type: "Sans Serif"
  },
  {
    name: "BIZ UDMincho",
    type: "Serif"
  },
  {
    name: "BIZ UDPGothic",
    type: "Sans Serif"
  },
  {
    name: "BIZ UDPMincho",
    type: "Serif"
  },
  {
    name: "Babylonica",
    type: "Handwriting"
  },
  {
    name: "Bacasime Antique",
    type: "Serif"
  },
  {
    name: "Bad Script",
    type: "Handwriting"
  },
  {
    name: "Bagel Fat One",
    type: "Display"
  },
  {
    name: "Bahiana",
    type: "Display"
  },
  {
    name: "Bahianita",
    type: "Display"
  },
  {
    name: "Bai Jamjuree",
    type: "Sans Serif"
  },
  {
    name: "Bakbak One",
    type: "Display"
  },
  {
    name: "Ballet",
    type: "Handwriting"
  },
  {
    name: "Baloo 2",
    type: "Display"
  },
  {
    name: "Baloo Bhai 2",
    type: "Display"
  },
  {
    name: "Baloo Bhaijaan 2",
    type: "Display"
  },
  {
    name: "Baloo Bhaina 2",
    type: "Display"
  },
  {
    name: "Baloo Chettan 2",
    type: "Display"
  },
  {
    name: "Baloo Da 2",
    type: "Display"
  },
  {
    name: "Baloo Paaji 2",
    type: "Display"
  },
  {
    name: "Baloo Tamma 2",
    type: "Display"
  },
  {
    name: "Baloo Tammudu 2",
    type: "Display"
  },
  {
    name: "Baloo Thambi 2",
    type: "Display"
  },
  {
    name: "Balsamiq Sans",
    type: "Display"
  },
  {
    name: "Balthazar",
    type: "Serif"
  },
  {
    name: "Bangers",
    type: "Display"
  },
  {
    name: "Barlow",
    type: "Sans Serif"
  },
  {
    name: "Barlow Condensed",
    type: "Sans Serif"
  },
  {
    name: "Barlow Semi Condensed",
    type: "Sans Serif"
  },
  {
    name: "Barriecito",
    type: "Display"
  },
  {
    name: "Barrio",
    type: "Display"
  },
  {
    name: "Basic",
    type: "Sans Serif"
  },
  {
    name: "Baskervville",
    type: "Serif"
  },
  {
    name: "Battambang",
    type: "Display"
  },
  {
    name: "Baumans",
    type: "Display"
  },
  {
    name: "Bayon",
    type: "Sans Serif"
  },
  {
    name: "Be Vietnam Pro",
    type: "Sans Serif"
  },
  {
    name: "Beau Rivage",
    type: "Handwriting"
  },
  {
    name: "Bebas Neue",
    type: "Sans Serif"
  },
  {
    name: "Belanosima",
    type: "Sans Serif"
  },
  {
    name: "Belgrano",
    type: "Serif"
  },
  {
    name: "Bellefair",
    type: "Serif"
  },
  {
    name: "Belleza",
    type: "Sans Serif"
  },
  {
    name: "Bellota",
    type: "Display"
  },
  {
    name: "Bellota Text",
    type: "Display"
  },
  {
    name: "BenchNine",
    type: "Sans Serif"
  },
  {
    name: "Benne",
    type: "Serif"
  },
  {
    name: "Bentham",
    type: "Serif"
  },
  {
    name: "Berkshire Swash",
    type: "Handwriting"
  },
  {
    name: "Besley",
    type: "Serif"
  },
  {
    name: "Beth Ellen",
    type: "Handwriting"
  },
  {
    name: "Bevan",
    type: "Serif"
  },
  {
    name: "BhuTuka Expanded One",
    type: "Serif"
  },
  {
    name: "Big Shoulders Display",
    type: "Display"
  },
  {
    name: "Big Shoulders Inline Display",
    type: "Display"
  },
  {
    name: "Big Shoulders Inline Text",
    type: "Display"
  },
  {
    name: "Big Shoulders Stencil Display",
    type: "Display"
  },
  {
    name: "Big Shoulders Stencil Text",
    type: "Display"
  },
  {
    name: "Big Shoulders Text",
    type: "Display"
  },
  {
    name: "Bigelow Rules",
    type: "Display"
  },
  {
    name: "Bigshot One",
    type: "Display"
  },
  {
    name: "Bilbo",
    type: "Handwriting"
  },
  {
    name: "Bilbo Swash Caps",
    type: "Handwriting"
  },
  {
    name: "BioRhyme",
    type: "Serif"
  },
  {
    name: "BioRhyme Expanded",
    type: "Serif"
  },
  {
    name: "Birthstone",
    type: "Handwriting"
  },
  {
    name: "Birthstone Bounce",
    type: "Handwriting"
  },
  {
    name: "Biryani",
    type: "Sans Serif"
  },
  {
    name: "Bitter",
    type: "Serif"
  },
  {
    name: "Black And White Picture",
    type: "Display"
  },
  {
    name: "Black Han Sans",
    type: "Sans Serif"
  },
  {
    name: "Black Ops One",
    type: "Display"
  },
  {
    name: "Blaka",
    type: "Display"
  },
  {
    name: "Blaka Hollow",
    type: "Display"
  },
  {
    name: "Blaka Ink",
    type: "Display"
  },
  {
    name: "Blinker",
    type: "Sans Serif"
  },
  {
    name: "Bodoni Moda",
    type: "Serif"
  },
  {
    name: "Bokor",
    type: "Display"
  },
  {
    name: "Bona Nova",
    type: "Serif"
  },
  {
    name: "Bonbon",
    type: "Handwriting"
  },
  {
    name: "Bonheur Royale",
    type: "Handwriting"
  },
  {
    name: "Boogaloo",
    type: "Display"
  },
  {
    name: "Borel",
    type: "Handwriting"
  },
  {
    name: "Bowlby One",
    type: "Display"
  },
  {
    name: "Bowlby One SC",
    type: "Display"
  },
  {
    name: "Braah One",
    type: "Sans Serif"
  },
  {
    name: "Brawler",
    type: "Serif"
  },
  {
    name: "Bree Serif",
    type: "Serif"
  },
  {
    name: "Bricolage Grotesque",
    type: "Sans Serif"
  },
  {
    name: "Bruno Ace",
    type: "Display"
  },
  {
    name: "Bruno Ace SC",
    type: "Display"
  },
  {
    name: "Brygada 1918",
    type: "Serif"
  },
  {
    name: "Bubblegum Sans",
    type: "Display"
  },
  {
    name: "Bubbler One",
    type: "Sans Serif"
  },
  {
    name: "Buda",
    type: "Display"
  },
  {
    name: "Buenard",
    type: "Serif"
  },
  {
    name: "Bungee",
    type: "Display"
  },
  {
    name: "Bungee Hairline",
    type: "Display"
  },
  {
    name: "Bungee Inline",
    type: "Display"
  },
  {
    name: "Bungee Outline",
    type: "Display"
  },
  {
    name: "Bungee Shade",
    type: "Display"
  },
  {
    name: "Bungee Spice",
    type: "Display"
  },
  {
    name: "Butcherman",
    type: "Display"
  },
  {
    name: "Butterfly Kids",
    type: "Handwriting"
  },
  {
    name: "Cabin",
    type: "Sans Serif"
  },
  {
    name: "Cabin Condensed",
    type: "Sans Serif"
  },
  {
    name: "Cabin Sketch",
    type: "Display"
  },
  {
    name: "Caesar Dressing",
    type: "Display"
  },
  {
    name: "Cagliostro",
    type: "Sans Serif"
  },
  {
    name: "Cairo",
    type: "Sans Serif"
  },
  {
    name: "Cairo Play",
    type: "Sans Serif"
  },
  {
    name: "Caladea",
    type: "Serif"
  },
  {
    name: "Calistoga",
    type: "Display"
  },
  {
    name: "Calligraffitti",
    type: "Handwriting"
  },
  {
    name: "Cambay",
    type: "Sans Serif"
  },
  {
    name: "Cambo",
    type: "Serif"
  },
  {
    name: "Candal",
    type: "Sans Serif"
  },
  {
    name: "Cantarell",
    type: "Sans Serif"
  },
  {
    name: "Cantata One",
    type: "Serif"
  },
  {
    name: "Cantora One",
    type: "Sans Serif"
  },
  {
    name: "Caprasimo",
    type: "Display"
  },
  {
    name: "Capriola",
    type: "Sans Serif"
  },
  {
    name: "Caramel",
    type: "Handwriting"
  },
  {
    name: "Carattere",
    type: "Handwriting"
  },
  {
    name: "Cardo",
    type: "Serif"
  },
  {
    name: "Carlito",
    type: "Sans Serif"
  },
  {
    name: "Carme",
    type: "Sans Serif"
  },
  {
    name: "Carrois Gothic",
    type: "Sans Serif"
  },
  {
    name: "Carrois Gothic SC",
    type: "Sans Serif"
  },
  {
    name: "Carter One",
    type: "Display"
  },
  {
    name: "Castoro",
    type: "Serif"
  },
  {
    name: "Castoro Titling",
    type: "Display"
  },
  {
    name: "Catamaran",
    type: "Sans Serif"
  },
  {
    name: "Caudex",
    type: "Serif"
  },
  {
    name: "Caveat",
    type: "Handwriting"
  },
  {
    name: "Caveat Brush",
    type: "Handwriting"
  },
  {
    name: "Cedarville Cursive",
    type: "Handwriting"
  },
  {
    name: "Ceviche One",
    type: "Display"
  },
  {
    name: "Chakra Petch",
    type: "Sans Serif"
  },
  {
    name: "Changa",
    type: "Sans Serif"
  },
  {
    name: "Changa One",
    type: "Display"
  },
  {
    name: "Chango",
    type: "Display"
  },
  {
    name: "Charis SIL",
    type: "Serif"
  },
  {
    name: "Charm",
    type: "Handwriting"
  },
  {
    name: "Charmonman",
    type: "Handwriting"
  },
  {
    name: "Chathura",
    type: "Sans Serif"
  },
  {
    name: "Chau Philomene One",
    type: "Sans Serif"
  },
  {
    name: "Chela One",
    type: "Display"
  },
  {
    name: "Chelsea Market",
    type: "Display"
  },
  {
    name: "Chenla",
    type: "Display"
  },
  {
    name: "Cherish",
    type: "Handwriting"
  },
  {
    name: "Cherry Bomb One",
    type: "Display"
  },
  {
    name: "Cherry Cream Soda",
    type: "Display"
  },
  {
    name: "Cherry Swash",
    type: "Display"
  },
  {
    name: "Chewy",
    type: "Display"
  },
  {
    name: "Chicle",
    type: "Display"
  },
  {
    name: "Chilanka",
    type: "Handwriting"
  },
  {
    name: "Chivo",
    type: "Sans Serif"
  },
  {
    name: "Chivo Mono",
    type: "Monospace"
  },
  {
    name: "Chokokutai",
    type: "Display"
  },
  {
    name: "Chonburi",
    type: "Display"
  },
  {
    name: "Cinzel",
    type: "Serif"
  },
  {
    name: "Cinzel Decorative",
    type: "Display"
  },
  {
    name: "Clicker Script",
    type: "Handwriting"
  },
  {
    name: "Climate Crisis",
    type: "Display"
  },
  {
    name: "Coda",
    type: "Display"
  },
  {
    name: "Codystar",
    type: "Display"
  },
  {
    name: "Coiny",
    type: "Display"
  },
  {
    name: "Combo",
    type: "Display"
  },
  {
    name: "Comfortaa",
    type: "Display"
  },
  {
    name: "Comforter",
    type: "Handwriting"
  },
  {
    name: "Comforter Brush",
    type: "Handwriting"
  },
  {
    name: "Comic Neue",
    type: "Handwriting"
  },
  {
    name: "Coming Soon",
    type: "Handwriting"
  },
  {
    name: "Comme",
    type: "Sans Serif"
  },
  {
    name: "Commissioner",
    type: "Sans Serif"
  },
  {
    name: "Concert One",
    type: "Display"
  },
  {
    name: "Condiment",
    type: "Handwriting"
  },
  {
    name: "Content",
    type: "Display"
  },
  {
    name: "Contrail One",
    type: "Display"
  },
  {
    name: "Convergence",
    type: "Sans Serif"
  },
  {
    name: "Cookie",
    type: "Handwriting"
  },
  {
    name: "Copse",
    type: "Serif"
  },
  {
    name: "Corben",
    type: "Display"
  },
  {
    name: "Corinthia",
    type: "Handwriting"
  },
  {
    name: "Cormorant",
    type: "Serif"
  },
  {
    name: "Cormorant Garamond",
    type: "Serif"
  },
  {
    name: "Cormorant Infant",
    type: "Serif"
  },
  {
    name: "Cormorant SC",
    type: "Serif"
  },
  {
    name: "Cormorant Unicase",
    type: "Serif"
  },
  {
    name: "Cormorant Upright",
    type: "Serif"
  },
  {
    name: "Courgette",
    type: "Handwriting"
  },
  {
    name: "Courier Prime",
    type: "Monospace"
  },
  {
    name: "Cousine",
    type: "Monospace"
  },
  {
    name: "Coustard",
    type: "Serif"
  },
  {
    name: "Covered By Your Grace",
    type: "Handwriting"
  },
  {
    name: "Crafty Girls",
    type: "Handwriting"
  },
  {
    name: "Creepster",
    type: "Display"
  },
  {
    name: "Crete Round",
    type: "Serif"
  },
  {
    name: "Crimson Pro",
    type: "Serif"
  },
  {
    name: "Crimson Text",
    type: "Serif"
  },
  {
    name: "Croissant One",
    type: "Display"
  },
  {
    name: "Crushed",
    type: "Display"
  },
  {
    name: "Cuprum",
    type: "Sans Serif"
  },
  {
    name: "Cute Font",
    type: "Display"
  },
  {
    name: "Cutive",
    type: "Serif"
  },
  {
    name: "Cutive Mono",
    type: "Monospace"
  },
  {
    name: "DM Mono",
    type: "Monospace"
  },
  {
    name: "DM Sans",
    type: "Sans Serif"
  },
  {
    name: "DM Serif Display",
    type: "Serif"
  },
  {
    name: "DM Serif Text",
    type: "Serif"
  },
  {
    name: "Dai Banna SIL",
    type: "Serif"
  },
  {
    name: "Damion",
    type: "Handwriting"
  },
  {
    name: "Dancing Script",
    type: "Handwriting"
  },
  {
    name: "Dangrek",
    type: "Display"
  },
  {
    name: "Darker Grotesque",
    type: "Sans Serif"
  },
  {
    name: "Darumadrop One",
    type: "Display"
  },
  {
    name: "David Libre",
    type: "Serif"
  },
  {
    name: "Dawning of a New Day",
    type: "Handwriting"
  },
  {
    name: "Days One",
    type: "Sans Serif"
  },
  {
    name: "Dekko",
    type: "Handwriting"
  },
  {
    name: "Dela Gothic One",
    type: "Display"
  },
  {
    name: "Delicious Handrawn",
    type: "Handwriting"
  },
  {
    name: "Delius",
    type: "Handwriting"
  },
  {
    name: "Delius Swash Caps",
    type: "Handwriting"
  },
  {
    name: "Delius Unicase",
    type: "Handwriting"
  },
  {
    name: "Della Respira",
    type: "Serif"
  },
  {
    name: "Denk One",
    type: "Sans Serif"
  },
  {
    name: "Devonshire",
    type: "Handwriting"
  },
  {
    name: "Dhurjati",
    type: "Sans Serif"
  },
  {
    name: "Didact Gothic",
    type: "Sans Serif"
  },
  {
    name: "Diphylleia",
    type: "Serif"
  },
  {
    name: "Diplomata",
    type: "Display"
  },
  {
    name: "Diplomata SC",
    type: "Display"
  },
  {
    name: "Do Hyeon",
    type: "Sans Serif"
  },
  {
    name: "Dokdo",
    type: "Display"
  },
  {
    name: "Domine",
    type: "Serif"
  },
  {
    name: "Donegal One",
    type: "Serif"
  },
  {
    name: "Dongle",
    type: "Sans Serif"
  },
  {
    name: "Doppio One",
    type: "Sans Serif"
  },
  {
    name: "Dorsa",
    type: "Sans Serif"
  },
  {
    name: "Dosis",
    type: "Sans Serif"
  },
  {
    name: "DotGothic16",
    type: "Sans Serif"
  },
  {
    name: "Dr Sugiyama",
    type: "Handwriting"
  },
  {
    name: "Duru Sans",
    type: "Sans Serif"
  },
  {
    name: "DynaPuff",
    type: "Display"
  },
  {
    name: "Dynalight",
    type: "Display"
  },
  {
    name: "EB Garamond",
    type: "Serif"
  },
  {
    name: "Eagle Lake",
    type: "Handwriting"
  },
  {
    name: "East Sea Dokdo",
    type: "Handwriting"
  },
  {
    name: "Eater",
    type: "Display"
  },
  {
    name: "Economica",
    type: "Sans Serif"
  },
  {
    name: "Eczar",
    type: "Serif"
  },
  {
    name: "Edu NSW ACT Foundation",
    type: "Handwriting"
  },
  {
    name: "Edu QLD Beginner",
    type: "Handwriting"
  },
  {
    name: "Edu SA Beginner",
    type: "Handwriting"
  },
  {
    name: "Edu TAS Beginner",
    type: "Handwriting"
  },
  {
    name: "Edu VIC WA NT Beginner",
    type: "Handwriting"
  },
  {
    name: "El Messiri",
    type: "Sans Serif"
  },
  {
    name: "Electrolize",
    type: "Sans Serif"
  },
  {
    name: "Elsie",
    type: "Display"
  },
  {
    name: "Elsie Swash Caps",
    type: "Display"
  },
  {
    name: "Emblema One",
    type: "Display"
  },
  {
    name: "Emilys Candy",
    type: "Display"
  },
  {
    name: "Encode Sans",
    type: "Sans Serif"
  },
  {
    name: "Encode Sans Condensed",
    type: "Sans Serif"
  },
  {
    name: "Encode Sans Expanded",
    type: "Sans Serif"
  },
  {
    name: "Encode Sans SC",
    type: "Sans Serif"
  },
  {
    name: "Encode Sans Semi Condensed",
    type: "Sans Serif"
  },
  {
    name: "Encode Sans Semi Expanded",
    type: "Sans Serif"
  },
  {
    name: "Engagement",
    type: "Handwriting"
  },
  {
    name: "Englebert",
    type: "Sans Serif"
  },
  {
    name: "Enriqueta",
    type: "Serif"
  },
  {
    name: "Ephesis",
    type: "Handwriting"
  },
  {
    name: "Epilogue",
    type: "Sans Serif"
  },
  {
    name: "Erica One",
    type: "Display"
  },
  {
    name: "Esteban",
    type: "Serif"
  },
  {
    name: "Estonia",
    type: "Handwriting"
  },
  {
    name: "Euphoria Script",
    type: "Handwriting"
  },
  {
    name: "Ewert",
    type: "Display"
  },
  {
    name: "Exo",
    type: "Sans Serif"
  },
  {
    name: "Exo 2",
    type: "Sans Serif"
  },
  {
    name: "Expletus Sans",
    type: "Display"
  },
  {
    name: "Explora",
    type: "Handwriting"
  },
  {
    name: "Fahkwang",
    type: "Sans Serif"
  },
  {
    name: "Familjen Grotesk",
    type: "Sans Serif"
  },
  {
    name: "Fanwood Text",
    type: "Serif"
  },
  {
    name: "Farro",
    type: "Sans Serif"
  },
  {
    name: "Farsan",
    type: "Display"
  },
  {
    name: "Fascinate",
    type: "Display"
  },
  {
    name: "Fascinate Inline",
    type: "Display"
  },
  {
    name: "Faster One",
    type: "Display"
  },
  {
    name: "Fasthand",
    type: "Display"
  },
  {
    name: "Fauna One",
    type: "Serif"
  },
  {
    name: "Faustina",
    type: "Serif"
  },
  {
    name: "Federant",
    type: "Display"
  },
  {
    name: "Federo",
    type: "Sans Serif"
  },
  {
    name: "Felipa",
    type: "Handwriting"
  },
  {
    name: "Fenix",
    type: "Serif"
  },
  {
    name: "Festive",
    type: "Handwriting"
  },
  {
    name: "Figtree",
    type: "Sans Serif"
  },
  {
    name: "Finger Paint",
    type: "Display"
  },
  {
    name: "Finlandica",
    type: "Sans Serif"
  },
  {
    name: "Fira Code",
    type: "Monospace"
  },
  {
    name: "Fira Mono",
    type: "Monospace"
  },
  {
    name: "Fira Sans",
    type: "Sans Serif"
  },
  {
    name: "Fira Sans Condensed",
    type: "Sans Serif"
  },
  {
    name: "Fira Sans Extra Condensed",
    type: "Sans Serif"
  },
  {
    name: "Fjalla One",
    type: "Sans Serif"
  },
  {
    name: "Fjord One",
    type: "Serif"
  },
  {
    name: "Flamenco",
    type: "Display"
  },
  {
    name: "Flavors",
    type: "Display"
  },
  {
    name: "Fleur De Leah",
    type: "Handwriting"
  },
  {
    name: "Flow Block",
    type: "Display"
  },
  {
    name: "Flow Circular",
    type: "Display"
  },
  {
    name: "Flow Rounded",
    type: "Display"
  },
  {
    name: "Foldit",
    type: "Display"
  },
  {
    name: "Fondamento",
    type: "Handwriting"
  },
  {
    name: "Fontdiner Swanky",
    type: "Display"
  },
  {
    name: "Forum",
    type: "Display"
  },
  {
    name: "Fragment Mono",
    type: "Monospace"
  },
  {
    name: "Francois One",
    type: "Sans Serif"
  },
  {
    name: "Frank Ruhl Libre",
    type: "Serif"
  },
  {
    name: "Fraunces",
    type: "Serif"
  },
  {
    name: "Freckle Face",
    type: "Display"
  },
  {
    name: "Fredericka the Great",
    type: "Display"
  },
  {
    name: "Fredoka",
    type: "Sans Serif"
  },
  {
    name: "Freehand",
    type: "Display"
  },
  {
    name: "Fresca",
    type: "Sans Serif"
  },
  {
    name: "Frijole",
    type: "Display"
  },
  {
    name: "Fruktur",
    type: "Display"
  },
  {
    name: "Fugaz One",
    type: "Display"
  },
  {
    name: "Fuggles",
    type: "Handwriting"
  },
  {
    name: "Fuzzy Bubbles",
    type: "Handwriting"
  },
  {
    name: "GFS Didot",
    type: "Serif"
  },
  {
    name: "GFS Neohellenic",
    type: "Sans Serif"
  },
  {
    name: "Gabarito",
    type: "Display"
  },
  {
    name: "Gabriela",
    type: "Serif"
  },
  {
    name: "Gaegu",
    type: "Handwriting"
  },
  {
    name: "Gafata",
    type: "Sans Serif"
  },
  {
    name: "Gajraj One",
    type: "Display"
  },
  {
    name: "Galada",
    type: "Display"
  },
  {
    name: "Galdeano",
    type: "Sans Serif"
  },
  {
    name: "Galindo",
    type: "Display"
  },
  {
    name: "Gamja Flower",
    type: "Handwriting"
  },
  {
    name: "Gantari",
    type: "Sans Serif"
  },
  {
    name: "Gasoek One",
    type: "Sans Serif"
  },
  {
    name: "Gayathri",
    type: "Sans Serif"
  },
  {
    name: "Gelasio",
    type: "Serif"
  },
  {
    name: "Gemunu Libre",
    type: "Sans Serif"
  },
  {
    name: "Genos",
    type: "Sans Serif"
  },
  {
    name: "Gentium Book Plus",
    type: "Serif"
  },
  {
    name: "Gentium Plus",
    type: "Serif"
  },
  {
    name: "Geo",
    type: "Sans Serif"
  },
  {
    name: "Geologica",
    type: "Sans Serif"
  },
  {
    name: "Georama",
    type: "Sans Serif"
  },
  {
    name: "Geostar",
    type: "Display"
  },
  {
    name: "Geostar Fill",
    type: "Display"
  },
  {
    name: "Germania One",
    type: "Display"
  },
  {
    name: "Gideon Roman",
    type: "Display"
  },
  {
    name: "Gidugu",
    type: "Sans Serif"
  },
  {
    name: "Gilda Display",
    type: "Serif"
  },
  {
    name: "Girassol",
    type: "Display"
  },
  {
    name: "Give You Glory",
    type: "Handwriting"
  },
  {
    name: "Glass Antiqua",
    type: "Display"
  },
  {
    name: "Glegoo",
    type: "Serif"
  },
  {
    name: "Gloock",
    type: "Serif"
  },
  {
    name: "Gloria Hallelujah",
    type: "Handwriting"
  },
  {
    name: "Glory",
    type: "Sans Serif"
  },
  {
    name: "Gluten",
    type: "Display"
  },
  {
    name: "Goblin One",
    type: "Display"
  },
  {
    name: "Gochi Hand",
    type: "Handwriting"
  },
  {
    name: "Goldman",
    type: "Display"
  },
  {
    name: "Golos Text",
    type: "Sans Serif"
  },
  {
    name: "Gorditas",
    type: "Display"
  },
  {
    name: "Gothic A1",
    type: "Sans Serif"
  },
  {
    name: "Gotu",
    type: "Sans Serif"
  },
  {
    name: "Goudy Bookletter 1911",
    type: "Serif"
  },
  {
    name: "Gowun Batang",
    type: "Serif"
  },
  {
    name: "Gowun Dodum",
    type: "Sans Serif"
  },
  {
    name: "Graduate",
    type: "Serif"
  },
  {
    name: "Grand Hotel",
    type: "Handwriting"
  },
  {
    name: "Grandiflora One",
    type: "Serif"
  },
  {
    name: "Grandstander",
    type: "Display"
  },
  {
    name: "Grape Nuts",
    type: "Handwriting"
  },
  {
    name: "Gravitas One",
    type: "Display"
  },
  {
    name: "Great Vibes",
    type: "Handwriting"
  },
  {
    name: "Grechen Fuemen",
    type: "Handwriting"
  },
  {
    name: "Grenze",
    type: "Serif"
  },
  {
    name: "Grenze Gotisch",
    type: "Display"
  },
  {
    name: "Grey Qo",
    type: "Handwriting"
  },
  {
    name: "Griffy",
    type: "Display"
  },
  {
    name: "Gruppo",
    type: "Sans Serif"
  },
  {
    name: "Gudea",
    type: "Sans Serif"
  },
  {
    name: "Gugi",
    type: "Display"
  },
  {
    name: "Gulzar",
    type: "Serif"
  },
  {
    name: "Gupter",
    type: "Serif"
  },
  {
    name: "Gurajada",
    type: "Serif"
  },
  {
    name: "Gwendolyn",
    type: "Handwriting"
  },
  {
    name: "Habibi",
    type: "Serif"
  },
  {
    name: "Hachi Maru Pop",
    type: "Handwriting"
  },
  {
    name: "Hahmlet",
    type: "Serif"
  },
  {
    name: "Halant",
    type: "Serif"
  },
  {
    name: "Hammersmith One",
    type: "Sans Serif"
  },
  {
    name: "Hanalei",
    type: "Display"
  },
  {
    name: "Hanalei Fill",
    type: "Display"
  },
  {
    name: "Handjet",
    type: "Display"
  },
  {
    name: "Handlee",
    type: "Handwriting"
  },
  {
    name: "Hanken Grotesk",
    type: "Sans Serif"
  },
  {
    name: "Hanuman",
    type: "Serif"
  },
  {
    name: "Happy Monkey",
    type: "Display"
  },
  {
    name: "Harmattan",
    type: "Sans Serif"
  },
  {
    name: "Headland One",
    type: "Serif"
  },
  {
    name: "Hedvig Letters Sans",
    type: "Sans Serif"
  },
  {
    name: "Hedvig Letters Serif",
    type: "Serif"
  },
  {
    name: "Heebo",
    type: "Sans Serif"
  },
  {
    name: "Henny Penny",
    type: "Display"
  },
  {
    name: "Hepta Slab",
    type: "Serif"
  },
  {
    name: "Herr Von Muellerhoff",
    type: "Handwriting"
  },
  {
    name: "Hi Melody",
    type: "Handwriting"
  },
  {
    name: "Hina Mincho",
    type: "Serif"
  },
  {
    name: "Hind",
    type: "Sans Serif"
  },
  {
    name: "Hind Guntur",
    type: "Sans Serif"
  },
  {
    name: "Hind Madurai",
    type: "Sans Serif"
  },
  {
    name: "Hind Siliguri",
    type: "Sans Serif"
  },
  {
    name: "Hind Vadodara",
    type: "Sans Serif"
  },
  {
    name: "Holtwood One SC",
    type: "Serif"
  },
  {
    name: "Homemade Apple",
    type: "Handwriting"
  },
  {
    name: "Homenaje",
    type: "Sans Serif"
  },
  {
    name: "Hubballi",
    type: "Sans Serif"
  },
  {
    name: "Hurricane",
    type: "Handwriting"
  },
  {
    name: "IBM Plex Mono",
    type: "Monospace"
  },
  {
    name: "IBM Plex Sans",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Arabic",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Condensed",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Devanagari",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Hebrew",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans JP",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans KR",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Thai",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Sans Thai Looped",
    type: "Sans Serif"
  },
  {
    name: "IBM Plex Serif",
    type: "Serif"
  },
  {
    name: "IM Fell DW Pica",
    type: "Serif"
  },
  {
    name: "IM Fell DW Pica SC",
    type: "Serif"
  },
  {
    name: "IM Fell Double Pica",
    type: "Serif"
  },
  {
    name: "IM Fell Double Pica SC",
    type: "Serif"
  },
  {
    name: "IM Fell English",
    type: "Serif"
  },
  {
    name: "IM Fell English SC",
    type: "Serif"
  },
  {
    name: "IM Fell French Canon",
    type: "Serif"
  },
  {
    name: "IM Fell French Canon SC",
    type: "Serif"
  },
  {
    name: "IM Fell Great Primer",
    type: "Serif"
  },
  {
    name: "IM Fell Great Primer SC",
    type: "Serif"
  },
  {
    name: "Ibarra Real Nova",
    type: "Serif"
  },
  {
    name: "Iceberg",
    type: "Display"
  },
  {
    name: "Iceland",
    type: "Display"
  },
  {
    name: "Imbue",
    type: "Serif"
  },
  {
    name: "Imperial Script",
    type: "Handwriting"
  },
  {
    name: "Imprima",
    type: "Sans Serif"
  },
  {
    name: "Inclusive Sans",
    type: "Sans Serif"
  },
  {
    name: "Inconsolata",
    type: "Monospace"
  },
  {
    name: "Inder",
    type: "Sans Serif"
  },
  {
    name: "Indie Flower",
    type: "Handwriting"
  },
  {
    name: "Ingrid Darling",
    type: "Handwriting"
  },
  {
    name: "Inika",
    type: "Serif"
  },
  {
    name: "Inknut Antiqua",
    type: "Serif"
  },
  {
    name: "Inria Sans",
    type: "Sans Serif"
  },
  {
    name: "Inria Serif",
    type: "Serif"
  },
  {
    name: "Inspiration",
    type: "Handwriting"
  },
  {
    name: "Instrument Sans",
    type: "Sans Serif"
  },
  {
    name: "Instrument Serif",
    type: "Serif"
  },
  {
    name: "Inter",
    type: "Sans Serif"
  },
  {
    name: "Inter Tight",
    type: "Sans Serif"
  },
  {
    name: "Irish Grover",
    type: "Display"
  },
  {
    name: "Island Moments",
    type: "Handwriting"
  },
  {
    name: "Istok Web",
    type: "Sans Serif"
  },
  {
    name: "Italiana",
    type: "Serif"
  },
  {
    name: "Italianno",
    type: "Handwriting"
  },
  {
    name: "Itim",
    type: "Handwriting"
  },
  {
    name: "Jacques Francois",
    type: "Serif"
  },
  {
    name: "Jacques Francois Shadow",
    type: "Display"
  },
  {
    name: "Jaldi",
    type: "Sans Serif"
  },
  {
    name: "JetBrains Mono",
    type: "Monospace"
  },
  {
    name: "Jim Nightshade",
    type: "Handwriting"
  },
  {
    name: "Joan",
    type: "Serif"
  },
  {
    name: "Jockey One",
    type: "Sans Serif"
  },
  {
    name: "Jolly Lodger",
    type: "Display"
  },
  {
    name: "Jomhuria",
    type: "Display"
  },
  {
    name: "Jomolhari",
    type: "Serif"
  },
  {
    name: "Josefin Sans",
    type: "Sans Serif"
  },
  {
    name: "Josefin Slab",
    type: "Serif"
  },
  {
    name: "Jost",
    type: "Sans Serif"
  },
  {
    name: "Joti One",
    type: "Display"
  },
  {
    name: "Jua",
    type: "Sans Serif"
  },
  {
    name: "Judson",
    type: "Serif"
  },
  {
    name: "Julee",
    type: "Handwriting"
  },
  {
    name: "Julius Sans One",
    type: "Sans Serif"
  },
  {
    name: "Junge",
    type: "Serif"
  },
  {
    name: "Jura",
    type: "Sans Serif"
  },
  {
    name: "Just Another Hand",
    type: "Handwriting"
  },
  {
    name: "Just Me Again Down Here",
    type: "Handwriting"
  },
  {
    name: "K2D",
    type: "Sans Serif"
  },
  {
    name: "Kablammo",
    type: "Display"
  },
  {
    name: "Kadwa",
    type: "Serif"
  },
  {
    name: "Kaisei Decol",
    type: "Serif"
  },
  {
    name: "Kaisei HarunoUmi",
    type: "Serif"
  },
  {
    name: "Kaisei Opti",
    type: "Serif"
  },
  {
    name: "Kaisei Tokumin",
    type: "Serif"
  },
  {
    name: "Kalam",
    type: "Handwriting"
  },
  {
    name: "Kalnia",
    type: "Serif"
  },
  {
    name: "Kameron",
    type: "Serif"
  },
  {
    name: "Kanit",
    type: "Sans Serif"
  },
  {
    name: "Kantumruy Pro",
    type: "Sans Serif"
  },
  {
    name: "Karantina",
    type: "Display"
  },
  {
    name: "Karla",
    type: "Sans Serif"
  },
  {
    name: "Karma",
    type: "Serif"
  },
  {
    name: "Katibeh",
    type: "Display"
  },
  {
    name: "Kaushan Script",
    type: "Handwriting"
  },
  {
    name: "Kavivanar",
    type: "Handwriting"
  },
  {
    name: "Kavoon",
    type: "Display"
  },
  {
    name: "Kay Pho Du",
    type: "Serif"
  },
  {
    name: "Kdam Thmor Pro",
    type: "Sans Serif"
  },
  {
    name: "Keania One",
    type: "Display"
  },
  {
    name: "Kelly Slab",
    type: "Display"
  },
  {
    name: "Kenia",
    type: "Display"
  },
  {
    name: "Khand",
    type: "Sans Serif"
  },
  {
    name: "Khmer",
    type: "Sans Serif"
  },
  {
    name: "Khula",
    type: "Sans Serif"
  },
  {
    name: "Kings",
    type: "Handwriting"
  },
  {
    name: "Kirang Haerang",
    type: "Display"
  },
  {
    name: "Kite One",
    type: "Sans Serif"
  },
  {
    name: "Kiwi Maru",
    type: "Serif"
  },
  {
    name: "Klee One",
    type: "Handwriting"
  },
  {
    name: "Knewave",
    type: "Display"
  },
  {
    name: "KoHo",
    type: "Sans Serif"
  },
  {
    name: "Kodchasan",
    type: "Sans Serif"
  },
  {
    name: "Koh Santepheap",
    type: "Serif"
  },
  {
    name: "Kolker Brush",
    type: "Handwriting"
  },
  {
    name: "Konkhmer Sleokchher",
    type: "Display"
  },
  {
    name: "Kosugi",
    type: "Sans Serif"
  },
  {
    name: "Kosugi Maru",
    type: "Sans Serif"
  },
  {
    name: "Kotta One",
    type: "Serif"
  },
  {
    name: "Koulen",
    type: "Display"
  },
  {
    name: "Kranky",
    type: "Display"
  },
  {
    name: "Kreon",
    type: "Serif"
  },
  {
    name: "Kristi",
    type: "Handwriting"
  },
  {
    name: "Krona One",
    type: "Sans Serif"
  },
  {
    name: "Krub",
    type: "Sans Serif"
  },
  {
    name: "Kufam",
    type: "Sans Serif"
  },
  {
    name: "Kulim Park",
    type: "Sans Serif"
  },
  {
    name: "Kumar One",
    type: "Display"
  },
  {
    name: "Kumar One Outline",
    type: "Display"
  },
  {
    name: "Kumbh Sans",
    type: "Sans Serif"
  },
  {
    name: "Kurale",
    type: "Serif"
  },
  {
    name: "La Belle Aurore",
    type: "Handwriting"
  },
  {
    name: "Labrada",
    type: "Serif"
  },
  {
    name: "Lacquer",
    type: "Display"
  },
  {
    name: "Laila",
    type: "Sans Serif"
  },
  {
    name: "Lakki Reddy",
    type: "Handwriting"
  },
  {
    name: "Lalezar",
    type: "Display"
  },
  {
    name: "Lancelot",
    type: "Display"
  },
  {
    name: "Langar",
    type: "Display"
  },
  {
    name: "Lateef",
    type: "Serif"
  },
  {
    name: "Lato",
    type: "Sans Serif"
  },
  {
    name: "Lavishly Yours",
    type: "Handwriting"
  },
  {
    name: "League Gothic",
    type: "Sans Serif"
  },
  {
    name: "League Script",
    type: "Handwriting"
  },
  {
    name: "League Spartan",
    type: "Sans Serif"
  },
  {
    name: "Leckerli One",
    type: "Handwriting"
  },
  {
    name: "Ledger",
    type: "Serif"
  },
  {
    name: "Lekton",
    type: "Sans Serif"
  },
  {
    name: "Lemon",
    type: "Display"
  },
  {
    name: "Lemonada",
    type: "Display"
  },
  {
    name: "Lexend",
    type: "Sans Serif"
  },
  {
    name: "Lexend Deca",
    type: "Sans Serif"
  },
  {
    name: "Lexend Exa",
    type: "Sans Serif"
  },
  {
    name: "Lexend Giga",
    type: "Sans Serif"
  },
  {
    name: "Lexend Mega",
    type: "Sans Serif"
  },
  {
    name: "Lexend Peta",
    type: "Sans Serif"
  },
  {
    name: "Lexend Tera",
    type: "Sans Serif"
  },
  {
    name: "Lexend Zetta",
    type: "Sans Serif"
  },
  {
    name: "Libre Barcode 128",
    type: "Display"
  },
  {
    name: "Libre Barcode 128 Text",
    type: "Display"
  },
  {
    name: "Libre Barcode 39",
    type: "Display"
  },
  {
    name: "Libre Barcode 39 Extended",
    type: "Display"
  },
  {
    name: "Libre Barcode 39 Extended Text",
    type: "Display"
  },
  {
    name: "Libre Barcode 39 Text",
    type: "Display"
  },
  {
    name: "Libre Barcode EAN13 Text",
    type: "Display"
  },
  {
    name: "Libre Baskerville",
    type: "Serif"
  },
  {
    name: "Libre Bodoni",
    type: "Serif"
  },
  {
    name: "Libre Caslon Display",
    type: "Serif"
  },
  {
    name: "Libre Caslon Text",
    type: "Serif"
  },
  {
    name: "Libre Franklin",
    type: "Sans Serif"
  },
  {
    name: "Licorice",
    type: "Handwriting"
  },
  {
    name: "Life Savers",
    type: "Display"
  },
  {
    name: "Lilita One",
    type: "Display"
  },
  {
    name: "Lily Script One",
    type: "Display"
  },
  {
    name: "Limelight",
    type: "Display"
  },
  {
    name: "Linden Hill",
    type: "Serif"
  },
  {
    name: "Linefont",
    type: "Display"
  },
  {
    name: "Lisu Bosa",
    type: "Serif"
  },
  {
    name: "Literata",
    type: "Serif"
  },
  {
    name: "Liu Jian Mao Cao",
    type: "Handwriting"
  },
  {
    name: "Livvic",
    type: "Sans Serif"
  },
  {
    name: "Lobster",
    type: "Display"
  },
  {
    name: "Lobster Two",
    type: "Display"
  },
  {
    name: "Londrina Outline",
    type: "Display"
  },
  {
    name: "Londrina Shadow",
    type: "Display"
  },
  {
    name: "Londrina Sketch",
    type: "Display"
  },
  {
    name: "Londrina Solid",
    type: "Display"
  },
  {
    name: "Long Cang",
    type: "Handwriting"
  },
  {
    name: "Lora",
    type: "Serif"
  },
  {
    name: "Love Light",
    type: "Handwriting"
  },
  {
    name: "Love Ya Like A Sister",
    type: "Display"
  },
  {
    name: "Loved by the King",
    type: "Handwriting"
  },
  {
    name: "Lovers Quarrel",
    type: "Handwriting"
  },
  {
    name: "Luckiest Guy",
    type: "Display"
  },
  {
    name: "Lugrasimo",
    type: "Handwriting"
  },
  {
    name: "Lumanosimo",
    type: "Handwriting"
  },
  {
    name: "Lunasima",
    type: "Sans Serif"
  },
  {
    name: "Lusitana",
    type: "Serif"
  },
  {
    name: "Lustria",
    type: "Serif"
  },
  {
    name: "Luxurious Roman",
    type: "Display"
  },
  {
    name: "Luxurious Script",
    type: "Handwriting"
  },
  {
    name: "M PLUS 1",
    type: "Sans Serif"
  },
  {
    name: "M PLUS 1 Code",
    type: "Monospace"
  },
  {
    name: "M PLUS 1p",
    type: "Sans Serif"
  },
  {
    name: "M PLUS 2",
    type: "Sans Serif"
  },
  {
    name: "M PLUS Code Latin",
    type: "Sans Serif"
  },
  {
    name: "M PLUS Rounded 1c",
    type: "Sans Serif"
  },
  {
    name: "Ma Shan Zheng",
    type: "Handwriting"
  },
  {
    name: "Macondo",
    type: "Display"
  },
  {
    name: "Macondo Swash Caps",
    type: "Display"
  },
  {
    name: "Mada",
    type: "Sans Serif"
  },
  {
    name: "Magra",
    type: "Sans Serif"
  },
  {
    name: "Maiden Orange",
    type: "Serif"
  },
  {
    name: "Maitree",
    type: "Serif"
  },
  {
    name: "Major Mono Display",
    type: "Monospace"
  },
  {
    name: "Mako",
    type: "Sans Serif"
  },
  {
    name: "Mali",
    type: "Handwriting"
  },
  {
    name: "Mallanna",
    type: "Sans Serif"
  },
  {
    name: "Mandali",
    type: "Sans Serif"
  },
  {
    name: "Manjari",
    type: "Sans Serif"
  },
  {
    name: "Manrope",
    type: "Sans Serif"
  },
  {
    name: "Mansalva",
    type: "Handwriting"
  },
  {
    name: "Manuale",
    type: "Serif"
  },
  {
    name: "Marcellus",
    type: "Serif"
  },
  {
    name: "Marcellus SC",
    type: "Serif"
  },
  {
    name: "Marck Script",
    type: "Handwriting"
  },
  {
    name: "Margarine",
    type: "Display"
  },
  {
    name: "Marhey",
    type: "Display"
  },
  {
    name: "Markazi Text",
    type: "Serif"
  },
  {
    name: "Marko One",
    type: "Serif"
  },
  {
    name: "Marmelad",
    type: "Sans Serif"
  },
  {
    name: "Martel",
    type: "Serif"
  },
  {
    name: "Martel Sans",
    type: "Sans Serif"
  },
  {
    name: "Martian Mono",
    type: "Monospace"
  },
  {
    name: "Marvel",
    type: "Sans Serif"
  },
  {
    name: "Mate",
    type: "Serif"
  },
  {
    name: "Mate SC",
    type: "Serif"
  },
  {
    name: "Maven Pro",
    type: "Sans Serif"
  },
  {
    name: "McLaren",
    type: "Display"
  },
  {
    name: "Mea Culpa",
    type: "Handwriting"
  },
  {
    name: "Meddon",
    type: "Handwriting"
  },
  {
    name: "MedievalSharp",
    type: "Display"
  },
  {
    name: "Medula One",
    type: "Display"
  },
  {
    name: "Meera Inimai",
    type: "Sans Serif"
  },
  {
    name: "Megrim",
    type: "Display"
  },
  {
    name: "Meie Script",
    type: "Handwriting"
  },
  {
    name: "Meow Script",
    type: "Handwriting"
  },
  {
    name: "Merienda",
    type: "Handwriting"
  },
  {
    name: "Merriweather",
    type: "Serif"
  },
  {
    name: "Merriweather Sans",
    type: "Sans Serif"
  },
  {
    name: "Metal",
    type: "Display"
  },
  {
    name: "Metal Mania",
    type: "Display"
  },
  {
    name: "Metamorphous",
    type: "Display"
  },
  {
    name: "Metrophobic",
    type: "Sans Serif"
  },
  {
    name: "Michroma",
    type: "Sans Serif"
  },
  {
    name: "Milonga",
    type: "Display"
  },
  {
    name: "Miltonian",
    type: "Display"
  },
  {
    name: "Miltonian Tattoo",
    type: "Display"
  },
  {
    name: "Mina",
    type: "Sans Serif"
  },
  {
    name: "Mingzat",
    type: "Sans Serif"
  },
  {
    name: "Miniver",
    type: "Display"
  },
  {
    name: "Miriam Libre",
    type: "Sans Serif"
  },
  {
    name: "Mirza",
    type: "Serif"
  },
  {
    name: "Miss Fajardose",
    type: "Handwriting"
  },
  {
    name: "Mitr",
    type: "Sans Serif"
  },
  {
    name: "Mochiy Pop One",
    type: "Sans Serif"
  },
  {
    name: "Mochiy Pop P One",
    type: "Sans Serif"
  },
  {
    name: "Modak",
    type: "Display"
  },
  {
    name: "Modern Antiqua",
    type: "Display"
  },
  {
    name: "Mogra",
    type: "Display"
  },
  {
    name: "Mohave",
    type: "Sans Serif"
  },
  {
    name: "Moirai One",
    type: "Display"
  },
  {
    name: "Molengo",
    type: "Sans Serif"
  },
  {
    name: "Molle",
    type: "Handwriting"
  },
  {
    name: "Monda",
    type: "Sans Serif"
  },
  {
    name: "Monofett",
    type: "Monospace"
  },
  {
    name: "Monomaniac One",
    type: "Sans Serif"
  },
  {
    name: "Monoton",
    type: "Display"
  },
  {
    name: "Monsieur La Doulaise",
    type: "Handwriting"
  },
  {
    name: "Montaga",
    type: "Serif"
  },
  {
    name: "Montagu Slab",
    type: "Serif"
  },
  {
    name: "MonteCarlo",
    type: "Handwriting"
  },
  {
    name: "Montez",
    type: "Handwriting"
  },
  {
    name: "Montserrat",
    type: "Sans Serif"
  },
  {
    name: "Montserrat Alternates",
    type: "Sans Serif"
  },
  {
    name: "Montserrat Subrayada",
    type: "Sans Serif"
  },
  {
    name: "Moo Lah Lah",
    type: "Display"
  },
  {
    name: "Mooli",
    type: "Sans Serif"
  },
  {
    name: "Moon Dance",
    type: "Handwriting"
  },
  {
    name: "Moul",
    type: "Display"
  },
  {
    name: "Moulpali",
    type: "Sans Serif"
  },
  {
    name: "Mountains of Christmas",
    type: "Display"
  },
  {
    name: "Mouse Memoirs",
    type: "Sans Serif"
  },
  {
    name: "Mr Bedfort",
    type: "Handwriting"
  },
  {
    name: "Mr Dafoe",
    type: "Handwriting"
  },
  {
    name: "Mr De Haviland",
    type: "Handwriting"
  },
  {
    name: "Mrs Saint Delafield",
    type: "Handwriting"
  },
  {
    name: "Mrs Sheppards",
    type: "Handwriting"
  },
  {
    name: "Ms Madi",
    type: "Handwriting"
  },
  {
    name: "Mukta",
    type: "Sans Serif"
  },
  {
    name: "Mukta Mahee",
    type: "Sans Serif"
  },
  {
    name: "Mukta Malar",
    type: "Sans Serif"
  },
  {
    name: "Mukta Vaani",
    type: "Sans Serif"
  },
  {
    name: "Mulish",
    type: "Sans Serif"
  },
  {
    name: "Murecho",
    type: "Sans Serif"
  },
  {
    name: "MuseoModerno",
    type: "Display"
  },
  {
    name: "My Soul",
    type: "Handwriting"
  },
  {
    name: "Mynerve",
    type: "Handwriting"
  },
  {
    name: "Mystery Quest",
    type: "Display"
  },
  {
    name: "NTR",
    type: "Sans Serif"
  },
  {
    name: "Nabla",
    type: "Display"
  },
  {
    name: "Nanum Brush Script",
    type: "Handwriting"
  },
  {
    name: "Nanum Gothic",
    type: "Sans Serif"
  },
  {
    name: "Nanum Gothic Coding",
    type: "Handwriting"
  },
  {
    name: "Nanum Myeongjo",
    type: "Serif"
  },
  {
    name: "Nanum Pen Script",
    type: "Handwriting"
  },
  {
    name: "Narnoor",
    type: "Serif"
  },
  {
    name: "Neonderthaw",
    type: "Handwriting"
  },
  {
    name: "Nerko One",
    type: "Handwriting"
  },
  {
    name: "Neucha",
    type: "Handwriting"
  },
  {
    name: "Neuton",
    type: "Serif"
  },
  {
    name: "New Rocker",
    type: "Display"
  },
  {
    name: "New Tegomin",
    type: "Serif"
  },
  {
    name: "News Cycle",
    type: "Sans Serif"
  },
  {
    name: "Newsreader",
    type: "Serif"
  },
  {
    name: "Niconne",
    type: "Handwriting"
  },
  {
    name: "Niramit",
    type: "Sans Serif"
  },
  {
    name: "Nixie One",
    type: "Display"
  },
  {
    name: "Nobile",
    type: "Sans Serif"
  },
  {
    name: "Nokora",
    type: "Sans Serif"
  },
  {
    name: "Norican",
    type: "Handwriting"
  },
  {
    name: "Nosifer",
    type: "Display"
  },
  {
    name: "Notable",
    type: "Sans Serif"
  },
  {
    name: "Nothing You Could Do",
    type: "Handwriting"
  },
  {
    name: "Noticia Text",
    type: "Serif"
  },
  {
    name: "Noto Color Emoji",
    type: "Sans Serif"
  },
  {
    name: "Noto Emoji",
    type: "Sans Serif"
  },
  {
    name: "Noto Kufi Arabic",
    type: "Sans Serif"
  },
  {
    name: "Noto Music",
    type: "Sans Serif"
  },
  {
    name: "Noto Naskh Arabic",
    type: "Serif"
  },
  {
    name: "Noto Nastaliq Urdu",
    type: "Serif"
  },
  {
    name: "Noto Rashi Hebrew",
    type: "Serif"
  },
  {
    name: "Noto Sans",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Adlam",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Adlam Unjoined",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Anatolian Hieroglyphs",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Arabic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Armenian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Avestan",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Balinese",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Bamum",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Bassa Vah",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Batak",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Bengali",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Bhaiksuki",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Brahmi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Buginese",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Buhid",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Canadian Aboriginal",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Carian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Caucasian Albanian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Chakma",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Cham",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Cherokee",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Chorasmian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Coptic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Cuneiform",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Cypriot",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Cypro Minoan",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Deseret",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Devanagari",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Display",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Duployan",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Egyptian Hieroglyphs",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Elbasan",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Elymaic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Ethiopic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Georgian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Glagolitic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Gothic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Grantha",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Gujarati",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Gunjala Gondi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Gurmukhi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans HK",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Hanifi Rohingya",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Hanunoo",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Hatran",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Hebrew",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Imperial Aramaic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Indic Siyaq Numbers",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Inscriptional Pahlavi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Inscriptional Parthian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans JP",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Javanese",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans KR",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Kaithi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Kannada",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Kawi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Kayah Li",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Kharoshthi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Khmer",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Khojki",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Khudawadi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lao",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lao Looped",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lepcha",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Limbu",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Linear A",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Linear B",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lisu",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lycian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Lydian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mahajani",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Malayalam",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mandaic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Manichaean",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Marchen",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Masaram Gondi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Math",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mayan Numerals",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Medefaidrin",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Meetei Mayek",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mende Kikakui",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Meroitic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Miao",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Modi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mongolian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mono",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Mro",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Multani",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Myanmar",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans NKo",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans NKo Unjoined",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Nabataean",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Nag Mundari",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Nandinagari",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans New Tai Lue",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Newa",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Nushu",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Ogham",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Ol Chiki",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Hungarian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Italic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old North Arabian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Permic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Persian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Sogdian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old South Arabian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Old Turkic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Oriya",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Osage",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Osmanya",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Pahawh Hmong",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Palmyrene",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Pau Cin Hau",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Phags Pa",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Phoenician",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Psalter Pahlavi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Rejang",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Runic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans SC",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Samaritan",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Saurashtra",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Sharada",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Shavian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Siddham",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans SignWriting",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Sinhala",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Sogdian",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Sora Sompeng",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Soyombo",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Sundanese",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Syloti Nagri",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Symbols",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Symbols 2",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Syriac",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Syriac Eastern",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans TC",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tagalog",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tagbanwa",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tai Le",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tai Tham",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tai Viet",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Takri",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tamil",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tamil Supplement",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tangsa",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Telugu",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Thaana",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Thai",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Thai Looped",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tifinagh",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Tirhuta",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Ugaritic",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Vai",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Vithkuqi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Wancho",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Warang Citi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Yi",
    type: "Sans Serif"
  },
  {
    name: "Noto Sans Zanabazar Square",
    type: "Sans Serif"
  },
  {
    name: "Noto Serif",
    type: "Serif"
  },
  {
    name: "Noto Serif Ahom",
    type: "Serif"
  },
  {
    name: "Noto Serif Armenian",
    type: "Serif"
  },
  {
    name: "Noto Serif Balinese",
    type: "Serif"
  },
  {
    name: "Noto Serif Bengali",
    type: "Serif"
  },
  {
    name: "Noto Serif Devanagari",
    type: "Serif"
  },
  {
    name: "Noto Serif Display",
    type: "Serif"
  },
  {
    name: "Noto Serif Dogra",
    type: "Serif"
  },
  {
    name: "Noto Serif Ethiopic",
    type: "Serif"
  },
  {
    name: "Noto Serif Georgian",
    type: "Serif"
  },
  {
    name: "Noto Serif Grantha",
    type: "Serif"
  },
  {
    name: "Noto Serif Gujarati",
    type: "Serif"
  },
  {
    name: "Noto Serif Gurmukhi",
    type: "Serif"
  },
  {
    name: "Noto Serif HK",
    type: "Serif"
  },
  {
    name: "Noto Serif Hebrew",
    type: "Serif"
  },
  {
    name: "Noto Serif JP",
    type: "Serif"
  },
  {
    name: "Noto Serif KR",
    type: "Serif"
  },
  {
    name: "Noto Serif Kannada",
    type: "Serif"
  },
  {
    name: "Noto Serif Khitan Small Script",
    type: "Serif"
  },
  {
    name: "Noto Serif Khmer",
    type: "Serif"
  },
  {
    name: "Noto Serif Khojki",
    type: "Serif"
  },
  {
    name: "Noto Serif Lao",
    type: "Serif"
  },
  {
    name: "Noto Serif Makasar",
    type: "Serif"
  },
  {
    name: "Noto Serif Malayalam",
    type: "Serif"
  },
  {
    name: "Noto Serif Myanmar",
    type: "Serif"
  },
  {
    name: "Noto Serif NP Hmong",
    type: "Serif"
  },
  {
    name: "Noto Serif Old Uyghur",
    type: "Serif"
  },
  {
    name: "Noto Serif Oriya",
    type: "Serif"
  },
  {
    name: "Noto Serif Ottoman Siyaq",
    type: "Serif"
  },
  {
    name: "Noto Serif SC",
    type: "Serif"
  },
  {
    name: "Noto Serif Sinhala",
    type: "Serif"
  },
  {
    name: "Noto Serif TC",
    type: "Serif"
  },
  {
    name: "Noto Serif Tamil",
    type: "Serif"
  },
  {
    name: "Noto Serif Tangut",
    type: "Serif"
  },
  {
    name: "Noto Serif Telugu",
    type: "Serif"
  },
  {
    name: "Noto Serif Thai",
    type: "Serif"
  },
  {
    name: "Noto Serif Tibetan",
    type: "Serif"
  },
  {
    name: "Noto Serif Toto",
    type: "Serif"
  },
  {
    name: "Noto Serif Vithkuqi",
    type: "Serif"
  },
  {
    name: "Noto Serif Yezidi",
    type: "Serif"
  },
  {
    name: "Noto Traditional Nushu",
    type: "Sans Serif"
  },
  {
    name: "Nova Cut",
    type: "Display"
  },
  {
    name: "Nova Flat",
    type: "Display"
  },
  {
    name: "Nova Mono",
    type: "Monospace"
  },
  {
    name: "Nova Oval",
    type: "Display"
  },
  {
    name: "Nova Round",
    type: "Display"
  },
  {
    name: "Nova Script",
    type: "Display"
  },
  {
    name: "Nova Slim",
    type: "Display"
  },
  {
    name: "Nova Square",
    type: "Display"
  },
  {
    name: "Numans",
    type: "Sans Serif"
  },
  {
    name: "Nunito",
    type: "Sans Serif"
  },
  {
    name: "Nunito Sans",
    type: "Sans Serif"
  },
  {
    name: "Nuosu SIL",
    type: "Sans Serif"
  },
  {
    name: "Odibee Sans",
    type: "Display"
  },
  {
    name: "Odor Mean Chey",
    type: "Serif"
  },
  {
    name: "Offside",
    type: "Display"
  },
  {
    name: "Oi",
    type: "Display"
  },
  {
    name: "Old Standard TT",
    type: "Serif"
  },
  {
    name: "Oldenburg",
    type: "Display"
  },
  {
    name: "Ole",
    type: "Handwriting"
  },
  {
    name: "Oleo Script",
    type: "Display"
  },
  {
    name: "Oleo Script Swash Caps",
    type: "Display"
  },
  {
    name: "Onest",
    type: "Sans Serif"
  },
  {
    name: "Oooh Baby",
    type: "Handwriting"
  },
  {
    name: "Open Sans",
    type: "Sans Serif"
  },
  {
    name: "Oranienbaum",
    type: "Serif"
  },
  {
    name: "Orbit",
    type: "Sans Serif"
  },
  {
    name: "Orbitron",
    type: "Sans Serif"
  },
  {
    name: "Oregano",
    type: "Display"
  },
  {
    name: "Orelega One",
    type: "Display"
  },
  {
    name: "Orienta",
    type: "Sans Serif"
  },
  {
    name: "Original Surfer",
    type: "Display"
  },
  {
    name: "Oswald",
    type: "Sans Serif"
  },
  {
    name: "Outfit",
    type: "Sans Serif"
  },
  {
    name: "Over the Rainbow",
    type: "Handwriting"
  },
  {
    name: "Overlock",
    type: "Display"
  },
  {
    name: "Overlock SC",
    type: "Display"
  },
  {
    name: "Overpass",
    type: "Sans Serif"
  },
  {
    name: "Overpass Mono",
    type: "Monospace"
  },
  {
    name: "Ovo",
    type: "Serif"
  },
  {
    name: "Oxanium",
    type: "Display"
  },
  {
    name: "Oxygen",
    type: "Sans Serif"
  },
  {
    name: "Oxygen Mono",
    type: "Monospace"
  },
  {
    name: "PT Mono",
    type: "Monospace"
  },
  {
    name: "PT Sans",
    type: "Sans Serif"
  },
  {
    name: "PT Sans Caption",
    type: "Sans Serif"
  },
  {
    name: "PT Sans Narrow",
    type: "Sans Serif"
  },
  {
    name: "PT Serif",
    type: "Serif"
  },
  {
    name: "PT Serif Caption",
    type: "Serif"
  },
  {
    name: "Pacifico",
    type: "Handwriting"
  },
  {
    name: "Padauk",
    type: "Sans Serif"
  },
  {
    name: "Padyakke Expanded One",
    type: "Serif"
  },
  {
    name: "Palanquin",
    type: "Sans Serif"
  },
  {
    name: "Palanquin Dark",
    type: "Sans Serif"
  },
  {
    name: "Palette Mosaic",
    type: "Display"
  },
  {
    name: "Pangolin",
    type: "Handwriting"
  },
  {
    name: "Paprika",
    type: "Display"
  },
  {
    name: "Parisienne",
    type: "Handwriting"
  },
  {
    name: "Passero One",
    type: "Display"
  },
  {
    name: "Passion One",
    type: "Display"
  },
  {
    name: "Passions Conflict",
    type: "Handwriting"
  },
  {
    name: "Pathway Extreme",
    type: "Sans Serif"
  },
  {
    name: "Pathway Gothic One",
    type: "Sans Serif"
  },
  {
    name: "Patrick Hand",
    type: "Handwriting"
  },
  {
    name: "Patrick Hand SC",
    type: "Handwriting"
  },
  {
    name: "Pattaya",
    type: "Sans Serif"
  },
  {
    name: "Patua One",
    type: "Display"
  },
  {
    name: "Pavanam",
    type: "Sans Serif"
  },
  {
    name: "Paytone One",
    type: "Sans Serif"
  },
  {
    name: "Peddana",
    type: "Serif"
  },
  {
    name: "Peralta",
    type: "Serif"
  },
  {
    name: "Permanent Marker",
    type: "Handwriting"
  },
  {
    name: "Petemoss",
    type: "Handwriting"
  },
  {
    name: "Petit Formal Script",
    type: "Handwriting"
  },
  {
    name: "Petrona",
    type: "Serif"
  },
  {
    name: "Philosopher",
    type: "Sans Serif"
  },
  {
    name: "Phudu",
    type: "Display"
  },
  {
    name: "Piazzolla",
    type: "Serif"
  },
  {
    name: "Piedra",
    type: "Display"
  },
  {
    name: "Pinyon Script",
    type: "Handwriting"
  },
  {
    name: "Pirata One",
    type: "Display"
  },
  {
    name: "Pixelify Sans",
    type: "Display"
  },
  {
    name: "Plaster",
    type: "Display"
  },
  {
    name: "Play",
    type: "Sans Serif"
  },
  {
    name: "Playball",
    type: "Display"
  },
  {
    name: "Playfair",
    type: "Serif"
  },
  {
    name: "Playfair Display",
    type: "Serif"
  },
  {
    name: "Playfair Display SC",
    type: "Serif"
  },
  {
    name: "Playpen Sans",
    type: "Handwriting"
  },
  {
    name: "Plus Jakarta Sans",
    type: "Sans Serif"
  },
  {
    name: "Podkova",
    type: "Serif"
  },
  {
    name: "Poiret One",
    type: "Display"
  },
  {
    name: "Poller One",
    type: "Display"
  },
  {
    name: "Poltawski Nowy",
    type: "Serif"
  },
  {
    name: "Poly",
    type: "Serif"
  },
  {
    name: "Pompiere",
    type: "Display"
  },
  {
    name: "Pontano Sans",
    type: "Sans Serif"
  },
  {
    name: "Poor Story",
    type: "Display"
  },
  {
    name: "Poppins",
    type: "Sans Serif"
  },
  {
    name: "Port Lligat Sans",
    type: "Sans Serif"
  },
  {
    name: "Port Lligat Slab",
    type: "Serif"
  },
  {
    name: "Potta One",
    type: "Display"
  },
  {
    name: "Pragati Narrow",
    type: "Sans Serif"
  },
  {
    name: "Praise",
    type: "Handwriting"
  },
  {
    name: "Prata",
    type: "Serif"
  },
  {
    name: "Preahvihear",
    type: "Sans Serif"
  },
  {
    name: "Press Start 2P",
    type: "Display"
  },
  {
    name: "Pridi",
    type: "Serif"
  },
  {
    name: "Princess Sofia",
    type: "Handwriting"
  },
  {
    name: "Prociono",
    type: "Serif"
  },
  {
    name: "Prompt",
    type: "Sans Serif"
  },
  {
    name: "Prosto One",
    type: "Display"
  },
  {
    name: "Proza Libre",
    type: "Sans Serif"
  },
  {
    name: "Public Sans",
    type: "Sans Serif"
  },
  {
    name: "Puppies Play",
    type: "Handwriting"
  },
  {
    name: "Puritan",
    type: "Sans Serif"
  },
  {
    name: "Purple Purse",
    type: "Display"
  },
  {
    name: "Qahiri",
    type: "Sans Serif"
  },
  {
    name: "Quando",
    type: "Serif"
  },
  {
    name: "Quantico",
    type: "Sans Serif"
  },
  {
    name: "Quattrocento",
    type: "Serif"
  },
  {
    name: "Quattrocento Sans",
    type: "Sans Serif"
  },
  {
    name: "Questrial",
    type: "Sans Serif"
  },
  {
    name: "Quicksand",
    type: "Sans Serif"
  },
  {
    name: "Quintessential",
    type: "Handwriting"
  },
  {
    name: "Qwigley",
    type: "Handwriting"
  },
  {
    name: "Qwitcher Grypen",
    type: "Handwriting"
  },
  {
    name: "REM",
    type: "Sans Serif"
  },
  {
    name: "Racing Sans One",
    type: "Display"
  },
  {
    name: "Radio Canada",
    type: "Sans Serif"
  },
  {
    name: "Radley",
    type: "Serif"
  },
  {
    name: "Rajdhani",
    type: "Sans Serif"
  },
  {
    name: "Rakkas",
    type: "Display"
  },
  {
    name: "Raleway",
    type: "Sans Serif"
  },
  {
    name: "Raleway Dots",
    type: "Display"
  },
  {
    name: "Ramabhadra",
    type: "Sans Serif"
  },
  {
    name: "Ramaraja",
    type: "Serif"
  },
  {
    name: "Rambla",
    type: "Sans Serif"
  },
  {
    name: "Rammetto One",
    type: "Display"
  },
  {
    name: "Rampart One",
    type: "Display"
  },
  {
    name: "Ranchers",
    type: "Display"
  },
  {
    name: "Rancho",
    type: "Handwriting"
  },
  {
    name: "Ranga",
    type: "Display"
  },
  {
    name: "Rasa",
    type: "Serif"
  },
  {
    name: "Rationale",
    type: "Sans Serif"
  },
  {
    name: "Ravi Prakash",
    type: "Display"
  },
  {
    name: "Readex Pro",
    type: "Sans Serif"
  },
  {
    name: "Recursive",
    type: "Sans Serif"
  },
  {
    name: "Red Hat Display",
    type: "Sans Serif"
  },
  {
    name: "Red Hat Mono",
    type: "Monospace"
  },
  {
    name: "Red Hat Text",
    type: "Sans Serif"
  },
  {
    name: "Red Rose",
    type: "Display"
  },
  {
    name: "Redacted",
    type: "Display"
  },
  {
    name: "Redacted Script",
    type: "Display"
  },
  {
    name: "Redressed",
    type: "Handwriting"
  },
  {
    name: "Reem Kufi",
    type: "Sans Serif"
  },
  {
    name: "Reem Kufi Fun",
    type: "Sans Serif"
  },
  {
    name: "Reem Kufi Ink",
    type: "Sans Serif"
  },
  {
    name: "Reenie Beanie",
    type: "Handwriting"
  },
  {
    name: "Reggae One",
    type: "Display"
  },
  {
    name: "Rethink Sans",
    type: "Sans Serif"
  },
  {
    name: "Revalia",
    type: "Display"
  },
  {
    name: "Rhodium Libre",
    type: "Serif"
  },
  {
    name: "Ribeye",
    type: "Display"
  },
  {
    name: "Ribeye Marrow",
    type: "Display"
  },
  {
    name: "Righteous",
    type: "Display"
  },
  {
    name: "Risque",
    type: "Display"
  },
  {
    name: "Road Rage",
    type: "Display"
  },
  {
    name: "Roboto",
    type: "Sans Serif"
  },
  {
    name: "Roboto Condensed",
    type: "Sans Serif"
  },
  {
    name: "Roboto Flex",
    type: "Sans Serif"
  },
  {
    name: "Roboto Mono",
    type: "Monospace"
  },
  {
    name: "Roboto Serif",
    type: "Serif"
  },
  {
    name: "Roboto Slab",
    type: "Serif"
  },
  {
    name: "Rochester",
    type: "Handwriting"
  },
  {
    name: "Rock 3D",
    type: "Display"
  },
  {
    name: "Rock Salt",
    type: "Handwriting"
  },
  {
    name: "RocknRoll One",
    type: "Sans Serif"
  },
  {
    name: "Rokkitt",
    type: "Serif"
  },
  {
    name: "Romanesco",
    type: "Handwriting"
  },
  {
    name: "Ropa Sans",
    type: "Sans Serif"
  },
  {
    name: "Rosario",
    type: "Sans Serif"
  },
  {
    name: "Rosarivo",
    type: "Serif"
  },
  {
    name: "Rouge Script",
    type: "Handwriting"
  },
  {
    name: "Rowdies",
    type: "Display"
  },
  {
    name: "Rozha One",
    type: "Serif"
  },
  {
    name: "Rubik",
    type: "Sans Serif"
  },
  {
    name: "Rubik 80s Fade",
    type: "Display"
  },
  {
    name: "Rubik Beastly",
    type: "Display"
  },
  {
    name: "Rubik Broken Fax",
    type: "Display"
  },
  {
    name: "Rubik Bubbles",
    type: "Display"
  },
  {
    name: "Rubik Burned",
    type: "Display"
  },
  {
    name: "Rubik Dirt",
    type: "Display"
  },
  {
    name: "Rubik Distressed",
    type: "Display"
  },
  {
    name: "Rubik Doodle Shadow",
    type: "Display"
  },
  {
    name: "Rubik Doodle Triangles",
    type: "Display"
  },
  {
    name: "Rubik Gemstones",
    type: "Display"
  },
  {
    name: "Rubik Glitch",
    type: "Display"
  },
  {
    name: "Rubik Iso",
    type: "Display"
  },
  {
    name: "Rubik Lines",
    type: "Display"
  },
  {
    name: "Rubik Maps",
    type: "Display"
  },
  {
    name: "Rubik Marker Hatch",
    type: "Display"
  },
  {
    name: "Rubik Maze",
    type: "Display"
  },
  {
    name: "Rubik Microbe",
    type: "Display"
  },
  {
    name: "Rubik Mono One",
    type: "Sans Serif"
  },
  {
    name: "Rubik Moonrocks",
    type: "Display"
  },
  {
    name: "Rubik Pixels",
    type: "Display"
  },
  {
    name: "Rubik Puddles",
    type: "Display"
  },
  {
    name: "Rubik Scribble",
    type: "Display"
  },
  {
    name: "Rubik Spray Paint",
    type: "Display"
  },
  {
    name: "Rubik Storm",
    type: "Display"
  },
  {
    name: "Rubik Vinyl",
    type: "Display"
  },
  {
    name: "Rubik Wet Paint",
    type: "Display"
  },
  {
    name: "Ruda",
    type: "Sans Serif"
  },
  {
    name: "Rufina",
    type: "Serif"
  },
  {
    name: "Ruge Boogie",
    type: "Handwriting"
  },
  {
    name: "Ruluko",
    type: "Sans Serif"
  },
  {
    name: "Rum Raisin",
    type: "Sans Serif"
  },
  {
    name: "Ruslan Display",
    type: "Display"
  },
  {
    name: "Russo One",
    type: "Sans Serif"
  },
  {
    name: "Ruthie",
    type: "Handwriting"
  },
  {
    name: "Ruwudu",
    type: "Serif"
  },
  {
    name: "Rye",
    type: "Display"
  },
  {
    name: "STIX Two Text",
    type: "Serif"
  },
  {
    name: "Sacramento",
    type: "Handwriting"
  },
  {
    name: "Sahitya",
    type: "Serif"
  },
  {
    name: "Sail",
    type: "Display"
  },
  {
    name: "Saira",
    type: "Sans Serif"
  },
  {
    name: "Saira Condensed",
    type: "Sans Serif"
  },
  {
    name: "Saira Extra Condensed",
    type: "Sans Serif"
  },
  {
    name: "Saira Semi Condensed",
    type: "Sans Serif"
  },
  {
    name: "Saira Stencil One",
    type: "Display"
  },
  {
    name: "Salsa",
    type: "Display"
  },
  {
    name: "Sanchez",
    type: "Serif"
  },
  {
    name: "Sancreek",
    type: "Display"
  },
  {
    name: "Sansita",
    type: "Sans Serif"
  },
  {
    name: "Sansita Swashed",
    type: "Display"
  },
  {
    name: "Sarabun",
    type: "Sans Serif"
  },
  {
    name: "Sarala",
    type: "Sans Serif"
  },
  {
    name: "Sarina",
    type: "Display"
  },
  {
    name: "Sarpanch",
    type: "Sans Serif"
  },
  {
    name: "Sassy Frass",
    type: "Handwriting"
  },
  {
    name: "Satisfy",
    type: "Handwriting"
  },
  {
    name: "Sawarabi Gothic",
    type: "Sans Serif"
  },
  {
    name: "Sawarabi Mincho",
    type: "Serif"
  },
  {
    name: "Scada",
    type: "Sans Serif"
  },
  {
    name: "Scheherazade New",
    type: "Serif"
  },
  {
    name: "Schibsted Grotesk",
    type: "Sans Serif"
  },
  {
    name: "Schoolbell",
    type: "Handwriting"
  },
  {
    name: "Scope One",
    type: "Serif"
  },
  {
    name: "Seaweed Script",
    type: "Display"
  },
  {
    name: "Secular One",
    type: "Sans Serif"
  },
  {
    name: "Sedgwick Ave",
    type: "Handwriting"
  },
  {
    name: "Sedgwick Ave Display",
    type: "Handwriting"
  },
  {
    name: "Sen",
    type: "Sans Serif"
  },
  {
    name: "Send Flowers",
    type: "Handwriting"
  },
  {
    name: "Sevillana",
    type: "Display"
  },
  {
    name: "Seymour One",
    type: "Sans Serif"
  },
  {
    name: "Shadows Into Light",
    type: "Handwriting"
  },
  {
    name: "Shadows Into Light Two",
    type: "Handwriting"
  },
  {
    name: "Shalimar",
    type: "Handwriting"
  },
  {
    name: "Shantell Sans",
    type: "Display"
  },
  {
    name: "Shanti",
    type: "Sans Serif"
  },
  {
    name: "Share",
    type: "Sans Serif"
  },
  {
    name: "Share Tech",
    type: "Sans Serif"
  },
  {
    name: "Share Tech Mono",
    type: "Monospace"
  },
  {
    name: "Shippori Antique",
    type: "Sans Serif"
  },
  {
    name: "Shippori Antique B1",
    type: "Sans Serif"
  },
  {
    name: "Shippori Mincho",
    type: "Serif"
  },
  {
    name: "Shippori Mincho B1",
    type: "Serif"
  },
  {
    name: "Shizuru",
    type: "Display"
  },
  {
    name: "Shojumaru",
    type: "Display"
  },
  {
    name: "Short Stack",
    type: "Handwriting"
  },
  {
    name: "Shrikhand",
    type: "Display"
  },
  {
    name: "Siemreap",
    type: "Sans Serif"
  },
  {
    name: "Sigmar",
    type: "Display"
  },
  {
    name: "Sigmar One",
    type: "Display"
  },
  {
    name: "Signika",
    type: "Sans Serif"
  },
  {
    name: "Signika Negative",
    type: "Sans Serif"
  },
  {
    name: "Silkscreen",
    type: "Display"
  },
  {
    name: "Simonetta",
    type: "Display"
  },
  {
    name: "Single Day",
    type: "Display"
  },
  {
    name: "Sintony",
    type: "Sans Serif"
  },
  {
    name: "Sirin Stencil",
    type: "Display"
  },
  {
    name: "Six Caps",
    type: "Sans Serif"
  },
  {
    name: "Skranji",
    type: "Display"
  },
  {
    name: "Slabo 13px",
    type: "Serif"
  },
  {
    name: "Slabo 27px",
    type: "Serif"
  },
  {
    name: "Slackey",
    type: "Display"
  },
  {
    name: "Slackside One",
    type: "Handwriting"
  },
  {
    name: "Smokum",
    type: "Display"
  },
  {
    name: "Smooch",
    type: "Handwriting"
  },
  {
    name: "Smooch Sans",
    type: "Sans Serif"
  },
  {
    name: "Smythe",
    type: "Display"
  },
  {
    name: "Sniglet",
    type: "Display"
  },
  {
    name: "Snippet",
    type: "Sans Serif"
  },
  {
    name: "Snowburst One",
    type: "Display"
  },
  {
    name: "Sofadi One",
    type: "Display"
  },
  {
    name: "Sofia",
    type: "Handwriting"
  },
  {
    name: "Sofia Sans",
    type: "Sans Serif"
  },
  {
    name: "Sofia Sans Condensed",
    type: "Sans Serif"
  },
  {
    name: "Sofia Sans Extra Condensed",
    type: "Sans Serif"
  },
  {
    name: "Sofia Sans Semi Condensed",
    type: "Sans Serif"
  },
  {
    name: "Solitreo",
    type: "Handwriting"
  },
  {
    name: "Solway",
    type: "Serif"
  },
  {
    name: "Sometype Mono",
    type: "Monospace"
  },
  {
    name: "Song Myung",
    type: "Serif"
  },
  {
    name: "Sono",
    type: "Sans Serif"
  },
  {
    name: "Sonsie One",
    type: "Display"
  },
  {
    name: "Sora",
    type: "Sans Serif"
  },
  {
    name: "Sorts Mill Goudy",
    type: "Serif"
  },
  {
    name: "Source Code Pro",
    type: "Monospace"
  },
  {
    name: "Source Sans 3",
    type: "Sans Serif"
  },
  {
    name: "Source Serif 4",
    type: "Serif"
  },
  {
    name: "Space Grotesk",
    type: "Sans Serif"
  },
  {
    name: "Space Mono",
    type: "Monospace"
  },
  {
    name: "Special Elite",
    type: "Display"
  },
  {
    name: "Spectral",
    type: "Serif"
  },
  {
    name: "Spectral SC",
    type: "Serif"
  },
  {
    name: "Spicy Rice",
    type: "Display"
  },
  {
    name: "Spinnaker",
    type: "Sans Serif"
  },
  {
    name: "Spirax",
    type: "Display"
  },
  {
    name: "Splash",
    type: "Handwriting"
  },
  {
    name: "Spline Sans",
    type: "Sans Serif"
  },
  {
    name: "Spline Sans Mono",
    type: "Monospace"
  },
  {
    name: "Squada One",
    type: "Display"
  },
  {
    name: "Square Peg",
    type: "Handwriting"
  },
  {
    name: "Sree Krushnadevaraya",
    type: "Serif"
  },
  {
    name: "Sriracha",
    type: "Handwriting"
  },
  {
    name: "Srisakdi",
    type: "Display"
  },
  {
    name: "Staatliches",
    type: "Display"
  },
  {
    name: "Stalemate",
    type: "Handwriting"
  },
  {
    name: "Stalinist One",
    type: "Display"
  },
  {
    name: "Stardos Stencil",
    type: "Display"
  },
  {
    name: "Stick",
    type: "Sans Serif"
  },
  {
    name: "Stick No Bills",
    type: "Sans Serif"
  },
  {
    name: "Stint Ultra Condensed",
    type: "Serif"
  },
  {
    name: "Stint Ultra Expanded",
    type: "Serif"
  },
  {
    name: "Stoke",
    type: "Serif"
  },
  {
    name: "Strait",
    type: "Sans Serif"
  },
  {
    name: "Style Script",
    type: "Handwriting"
  },
  {
    name: "Stylish",
    type: "Sans Serif"
  },
  {
    name: "Sue Ellen Francisco",
    type: "Handwriting"
  },
  {
    name: "Suez One",
    type: "Serif"
  },
  {
    name: "Sulphur Point",
    type: "Sans Serif"
  },
  {
    name: "Sumana",
    type: "Serif"
  },
  {
    name: "Sunflower",
    type: "Sans Serif"
  },
  {
    name: "Sunshiney",
    type: "Handwriting"
  },
  {
    name: "Supermercado One",
    type: "Display"
  },
  {
    name: "Sura",
    type: "Serif"
  },
  {
    name: "Suranna",
    type: "Serif"
  },
  {
    name: "Suravaram",
    type: "Serif"
  },
  {
    name: "Suwannaphum",
    type: "Serif"
  },
  {
    name: "Swanky and Moo Moo",
    type: "Handwriting"
  },
  {
    name: "Syncopate",
    type: "Sans Serif"
  },
  {
    name: "Syne",
    type: "Sans Serif"
  },
  {
    name: "Syne Mono",
    type: "Monospace"
  },
  {
    name: "Syne Tactile",
    type: "Display"
  },
  {
    name: "Tai Heritage Pro",
    type: "Serif"
  },
  {
    name: "Tajawal",
    type: "Sans Serif"
  },
  {
    name: "Tangerine",
    type: "Handwriting"
  },
  {
    name: "Tapestry",
    type: "Handwriting"
  },
  {
    name: "Taprom",
    type: "Display"
  },
  {
    name: "Tauri",
    type: "Sans Serif"
  },
  {
    name: "Taviraj",
    type: "Serif"
  },
  {
    name: "Teko",
    type: "Sans Serif"
  },
  {
    name: "Tektur",
    type: "Display"
  },
  {
    name: "Telex",
    type: "Sans Serif"
  },
  {
    name: "Tenali Ramakrishna",
    type: "Sans Serif"
  },
  {
    name: "Tenor Sans",
    type: "Sans Serif"
  },
  {
    name: "Text Me One",
    type: "Sans Serif"
  },
  {
    name: "Texturina",
    type: "Serif"
  },
  {
    name: "Thasadith",
    type: "Sans Serif"
  },
  {
    name: "The Girl Next Door",
    type: "Handwriting"
  },
  {
    name: "The Nautigal",
    type: "Handwriting"
  },
  {
    name: "Tienne",
    type: "Serif"
  },
  {
    name: "Tillana",
    type: "Display"
  },
  {
    name: "Tilt Neon",
    type: "Display"
  },
  {
    name: "Tilt Prism",
    type: "Display"
  },
  {
    name: "Tilt Warp",
    type: "Display"
  },
  {
    name: "Timmana",
    type: "Sans Serif"
  },
  {
    name: "Tinos",
    type: "Serif"
  },
  {
    name: "Tiro Bangla",
    type: "Serif"
  },
  {
    name: "Tiro Devanagari Hindi",
    type: "Serif"
  },
  {
    name: "Tiro Devanagari Marathi",
    type: "Serif"
  },
  {
    name: "Tiro Devanagari Sanskrit",
    type: "Serif"
  },
  {
    name: "Tiro Gurmukhi",
    type: "Serif"
  },
  {
    name: "Tiro Kannada",
    type: "Serif"
  },
  {
    name: "Tiro Tamil",
    type: "Serif"
  },
  {
    name: "Tiro Telugu",
    type: "Serif"
  },
  {
    name: "Titan One",
    type: "Display"
  },
  {
    name: "Titillium Web",
    type: "Sans Serif"
  },
  {
    name: "Tomorrow",
    type: "Sans Serif"
  },
  {
    name: "Tourney",
    type: "Display"
  },
  {
    name: "Trade Winds",
    type: "Display"
  },
  {
    name: "Train One",
    type: "Display"
  },
  {
    name: "Trirong",
    type: "Serif"
  },
  {
    name: "Trispace",
    type: "Sans Serif"
  },
  {
    name: "Trocchi",
    type: "Serif"
  },
  {
    name: "Trochut",
    type: "Display"
  },
  {
    name: "Truculenta",
    type: "Sans Serif"
  },
  {
    name: "Trykker",
    type: "Serif"
  },
  {
    name: "Tsukimi Rounded",
    type: "Sans Serif"
  },
  {
    name: "Tulpen One",
    type: "Display"
  },
  {
    name: "Turret Road",
    type: "Display"
  },
  {
    name: "Twinkle Star",
    type: "Handwriting"
  },
  {
    name: "Ubuntu",
    type: "Sans Serif"
  },
  {
    name: "Ubuntu Condensed",
    type: "Sans Serif"
  },
  {
    name: "Ubuntu Mono",
    type: "Monospace"
  },
  {
    name: "Uchen",
    type: "Serif"
  },
  {
    name: "Ultra",
    type: "Serif"
  },
  {
    name: "Unbounded",
    type: "Sans Serif"
  },
  {
    name: "Uncial Antiqua",
    type: "Display"
  },
  {
    name: "Underdog",
    type: "Display"
  },
  {
    name: "Unica One",
    type: "Display"
  },
  {
    name: "UnifrakturCook",
    type: "Display"
  },
  {
    name: "UnifrakturMaguntia",
    type: "Display"
  },
  {
    name: "Unkempt",
    type: "Display"
  },
  {
    name: "Unlock",
    type: "Display"
  },
  {
    name: "Unna",
    type: "Serif"
  },
  {
    name: "Updock",
    type: "Handwriting"
  },
  {
    name: "Urbanist",
    type: "Sans Serif"
  },
  {
    name: "VT323",
    type: "Monospace"
  },
  {
    name: "Vampiro One",
    type: "Display"
  },
  {
    name: "Varela",
    type: "Sans Serif"
  },
  {
    name: "Varela Round",
    type: "Sans Serif"
  },
  {
    name: "Varta",
    type: "Sans Serif"
  },
  {
    name: "Vast Shadow",
    type: "Serif"
  },
  {
    name: "Vazirmatn",
    type: "Sans Serif"
  },
  {
    name: "Vesper Libre",
    type: "Serif"
  },
  {
    name: "Viaoda Libre",
    type: "Display"
  },
  {
    name: "Vibes",
    type: "Display"
  },
  {
    name: "Vibur",
    type: "Handwriting"
  },
  {
    name: "Victor Mono",
    type: "Monospace"
  },
  {
    name: "Vidaloka",
    type: "Serif"
  },
  {
    name: "Viga",
    type: "Sans Serif"
  },
  {
    name: "Vina Sans",
    type: "Display"
  },
  {
    name: "Voces",
    type: "Sans Serif"
  },
  {
    name: "Volkhov",
    type: "Serif"
  },
  {
    name: "Vollkorn",
    type: "Serif"
  },
  {
    name: "Vollkorn SC",
    type: "Serif"
  },
  {
    name: "Voltaire",
    type: "Sans Serif"
  },
  {
    name: "Vujahday Script",
    type: "Handwriting"
  },
  {
    name: "Waiting for the Sunrise",
    type: "Handwriting"
  },
  {
    name: "Wallpoet",
    type: "Display"
  },
  {
    name: "Walter Turncoat",
    type: "Handwriting"
  },
  {
    name: "Warnes",
    type: "Display"
  },
  {
    name: "Water Brush",
    type: "Handwriting"
  },
  {
    name: "Waterfall",
    type: "Handwriting"
  },
  {
    name: "Wavefont",
    type: "Display"
  },
  {
    name: "Wellfleet",
    type: "Serif"
  },
  {
    name: "Wendy One",
    type: "Sans Serif"
  },
  {
    name: "Whisper",
    type: "Handwriting"
  },
  {
    name: "WindSong",
    type: "Handwriting"
  },
  {
    name: "Wire One",
    type: "Sans Serif"
  },
  {
    name: "Wix Madefor Display",
    type: "Sans Serif"
  },
  {
    name: "Wix Madefor Text",
    type: "Sans Serif"
  },
  {
    name: "Work Sans",
    type: "Sans Serif"
  },
  {
    name: "Xanh Mono",
    type: "Monospace"
  },
  {
    name: "Yaldevi",
    type: "Sans Serif"
  },
  {
    name: "Yanone Kaffeesatz",
    type: "Sans Serif"
  },
  {
    name: "Yantramanav",
    type: "Sans Serif"
  },
  {
    name: "Yatra One",
    type: "Display"
  },
  {
    name: "Yellowtail",
    type: "Handwriting"
  },
  {
    name: "Yeon Sung",
    type: "Display"
  },
  {
    name: "Yeseva One",
    type: "Display"
  },
  {
    name: "Yesteryear",
    type: "Handwriting"
  },
  {
    name: "Yomogi",
    type: "Handwriting"
  },
  {
    name: "Young Serif",
    type: "Serif"
  },
  {
    name: "Yrsa",
    type: "Serif"
  },
  {
    name: "Ysabeau",
    type: "Sans Serif"
  },
  {
    name: "Ysabeau Infant",
    type: "Sans Serif"
  },
  {
    name: "Ysabeau Office",
    type: "Sans Serif"
  },
  {
    name: "Ysabeau SC",
    type: "Sans Serif"
  },
  {
    name: "Yuji Boku",
    type: "Serif"
  },
  {
    name: "Yuji Hentaigana Akari",
    type: "Handwriting"
  },
  {
    name: "Yuji Hentaigana Akebono",
    type: "Handwriting"
  },
  {
    name: "Yuji Mai",
    type: "Serif"
  },
  {
    name: "Yuji Syuku",
    type: "Serif"
  },
  {
    name: "Yusei Magic",
    type: "Sans Serif"
  },
  {
    name: "ZCOOL KuaiLe",
    type: "Sans Serif"
  },
  {
    name: "ZCOOL QingKe HuangYou",
    type: "Sans Serif"
  },
  {
    name: "ZCOOL XiaoWei",
    type: "Sans Serif"
  },
  {
    name: "Zen Antique",
    type: "Serif"
  },
  {
    name: "Zen Antique Soft",
    type: "Serif"
  },
  {
    name: "Zen Dots",
    type: "Display"
  },
  {
    name: "Zen Kaku Gothic Antique",
    type: "Sans Serif"
  },
  {
    name: "Zen Kaku Gothic New",
    type: "Sans Serif"
  },
  {
    name: "Zen Kurenaido",
    type: "Sans Serif"
  },
  {
    name: "Zen Loop",
    type: "Display"
  },
  {
    name: "Zen Maru Gothic",
    type: "Sans Serif"
  },
  {
    name: "Zen Old Mincho",
    type: "Serif"
  },
  {
    name: "Zen Tokyo Zoo",
    type: "Display"
  },
  {
    name: "Zeyada",
    type: "Handwriting"
  },
  {
    name: "Zhi Mang Xing",
    type: "Handwriting"
  },
  {
    name: "Zilla Slab",
    type: "Serif"
  },
  {
    name: "Zilla Slab Highlight",
    type: "Serif"
  }
], q = xe.filter((S) => !S.name.startsWith("Noto ")), Re = [...new Set(q.map((S) => S.type))], Ge = q.reduce((S, c) => (S[c.name] = c.type, S), {}), Fe = (S) => {
  switch (Ge[S]) {
    case "Serif":
      return "serif";
    case "Monospace":
      return "monospace";
    case "Handwriting":
      return "cursive";
    default:
      return "sans-serif";
  }
}, Ee = (S) => `https://fonts.google.com/share?selection.family=${S.replaceAll(" ", "+")}`, Ke = (S) => `<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=${S.replaceAll(" ", "+")}"/>`, Ie = (S) => `font-family: ${S}, ${Fe(S)};`, je = (S, c) => c ? `body * { font-family: '${S}' !important; }` : `body { font-family: '${S}' !important; }`, qe = (S, c) => [
  '<option value="">-- Select Font --</option>',
  ...(S ? q.filter((f) => f.type === S) : q).map((f) => `<option ${f.name === c ? "selected" : ""}>${f.name}</option>`)
], We = (S) => [
  '<option value="">All Types</option>',
  ...Re.map((c) => `<option ${c === S ? "selected" : ""}>${c}</option>`).join("")
], ze = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="500" zoomAndPan="magnify" viewBox="0 0 375 374.999991" height="500" preserveAspectRatio="xMidYMid meet" version="1.0"><defs><clipPath id="7cb730244f"><path d="M 0 46 L 272 46 L 272 329.4375 L 0 329.4375 Z M 0 46 " clip-rule="nonzero"/></clipPath></defs><path fill="#ffffff" d="M 313.511719 170.828125 L 287.996094 170.828125 L 248.988281 269.277344 L 263.609375 306.1875 L 271.289062 286.210938 L 330.484375 286.210938 L 345.730469 325.957031 L 375 325.957031 Z M 280.132812 263.105469 L 285.671875 248.675781 C 289.929688 237.761719 294.8125 224.019531 300.230469 207.796875 L 300.320312 207.53125 L 301.363281 207.53125 L 301.453125 207.796875 C 306.902344 223.929688 311.816406 237.671875 316.011719 248.675781 L 321.550781 263.105469 Z M 280.132812 263.105469 " fill-opacity="1" fill-rule="nonzero"/><g clip-path="url(#7cb730244f)"><path fill="#ffffff" d="M 262.953125 308.933594 L 262.742188 308.394531 L 248.09375 271.453125 L 247.886719 270.917969 L 247.675781 270.378906 L 158.765625 46.078125 L 112.253906 46.078125 L 0 329.296875 L 53.746094 329.296875 L 81.617188 256.695312 L 189.824219 256.695312 L 217.691406 329.296875 L 271.023438 329.296875 L 263.191406 309.5 Z M 97.785156 214.65625 L 107.9375 188.210938 C 115.710938 168.292969 124.644531 143.15625 134.527344 113.550781 L 134.707031 113.015625 L 136.703125 113.015625 L 136.878906 113.550781 C 146.855469 142.980469 155.789062 168.113281 163.46875 188.210938 L 173.625 214.65625 Z M 97.785156 214.65625 "  fill-opacity="1" fill-rule="nonzero"/></g></svg>', Ue = "Font Picker", oe = "astro:dev-overlay:font-picker", Ve = {
  id: "astro-font-picker",
  name: Ue,
  icon: ze,
  init(S, c) {
    const l = JSON.parse(sessionStorage.getItem(oe)) || { enabled: !0, override: !1 }, f = (d, C) => {
      l[d] = C, sessionStorage.setItem(oe, JSON.stringify(l));
    }, v = () => {
      c.dispatchEvent(
        new CustomEvent("toggle-notification", {
          detail: {
            state: !!l.font && l.enabled
          }
        })
      );
    }, A = document.createElement("style");
    A.setAttribute("id", "astro-font-picker");
    const h = document.createElement("astro-dev-toolbar-window");
    h.innerHTML = Te;
    const k = h.querySelector("#instr1"), O = h.querySelector("#instr2"), N = h.querySelector("#google-link"), D = h.querySelector("#toggle");
    D.input.checked = l.enabled ?? !0, D.input.addEventListener("change", () => {
      f("enabled", D.input.checked), D.input.checked ? b(l.font) : (document.querySelector("#astro-font-picker").remove(), v());
    });
    const B = h.querySelector("#override");
    B.input.checked = l.override ?? !0, B.input.addEventListener("change", () => {
      f("override", B.input.checked), l.enabled && b(l.font);
    });
    const b = (d) => {
      f("font", d), v(), d && l.enabled ? (Pe.load({
        google: {
          families: [d]
        },
        active: () => {
          A.textContent = je(d, l.override), N.href = Ee(d), k.innerText = Ke(d), O.innerText = Ie(d), N.style.visibility = "visible", k.style.visibility = "visible", O.style.visibility = "visible";
        }
      }), document.querySelector("#astro-font-picker") || document.head.appendChild(A)) : (A.textContent = "", N.style.visibility = "hidden", k.style.visibility = "hidden", O.style.visibility = "hidden");
    };
    l.enabled && b(l.font);
    const R = (d) => {
      P.innerHTML = qe(d, l.font).join(""), P.querySelector("option[selected]") || b("");
    }, L = h.querySelector("#group-select");
    L.innerHTML = We(l.group).join(""), L.addEventListener("change", (d) => {
      const G = d.target.value;
      f("group", G), R(G);
    });
    const P = h.querySelector("#font-select");
    R(l.group), P.addEventListener("change", (d) => {
      const C = d.target;
      b(C.value);
    }), S.appendChild(h);
  }
};
export {
  Ve as default
};
